<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>FAMR Simulations</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="site_libs/viz-1.8.2/viz.js"></script>
<link href="site_libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="site_libs/grViz-binding-1.0.10/grViz.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mvmr_website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="simulations.html">Old Simulations</a>
</li>
<li>
  <a href="famrsims.html">New FAMR Simulations</a>
</li>
<li>
  <a href="experimental.html">Old Experiments</a>
</li>
<li>
  <a href="realdata.html">Metabolite MR</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nlapier2/mvmr_website">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FAMR Simulations</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mvmr_website/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221205code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221205code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221205)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree7800bcd959a68adcd33803da73934c50f2421e76targetblank7800bcda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nlapier2/mvmr_website/tree/7800bcd959a68adcd33803da73934c50f2421e76" target="_blank">7800bcd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree7800bcd959a68adcd33803da73934c50f2421e76targetblank7800bcda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nlapier2/mvmr_website/tree/7800bcd959a68adcd33803da73934c50f2421e76" target="_blank">7800bcd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/famrsims.Rmd
    Modified:   data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt
    Modified:   data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_ivw_gfa.txt
    Modified:   data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/performance_famr_gfa.txt
    Modified:   data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/performance_ivw_gfa.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/famrsims.Rmd</code>) and HTML
(<code>docs/famrsims.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/analysis/famrsims.Rmd" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
<td>
added rsparsepro and rescale_z sim results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/famrsims.html" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
<td>
added rsparsepro and rescale_z sim results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/analysis/famrsims.Rmd" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
<td>
added results with new factor impute + high prior pi approach
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/famrsims.html" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
<td>
added results with new factor impute + high prior pi approach
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/373c87c39e78383ba04232a6cffb126a8d284f3e/analysis/famrsims.Rmd" target="_blank">373c87c</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-08-09
</td>
<td>
added state of the project, as well as details on real data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/373c87c39e78383ba04232a6cffb126a8d284f3e/docs/famrsims.html" target="_blank">373c87c</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-08-09
</td>
<td>
added state of the project, as well as details on real data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/analysis/famrsims.Rmd" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
<td>
added new FAMR simulations, and updated real data results with
annihilator mode
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/famrsims.html" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
<td>
added new FAMR simulations, and updated real data results with
annihilator mode
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/32197cac9632f9a9422643ee2533f3660a261591/analysis/famrsims.Rmd" target="_blank">32197ca</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-01
</td>
<td>
started working on adding new FAMR sims with annihilated factors
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggpubr&#39;:

    get_legend</code></pre>
<div id="simulation-description" class="section level2">
<h2>Simulation description</h2>
<p>We simulate according to the following DAG:</p>
<div id="htmlwidget-e3d5eff22888eb7ba78a" style="width:800px;height:350px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-e3d5eff22888eb7ba78a">{"x":{"diagram":"digraph {\n  graph [layout = dot, rankdir = LR]\n\n  node [shape = circle]\n  G [label = \"G\"]\n  Z [label = \"Z\"]\n  X [label = \"X\"]\n  Y [label = \"Y\"]\n\n  # edge definitions with the node IDs\n  edge []\n  G -> X [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GX<\/SUB><\/FONT>>]\n  G -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GY<\/SUB><\/FONT>>]\n  G -> Z [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GZ<\/SUB><\/FONT>>]\n  X -> Z [dir=back; label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZX<\/SUB><\/FONT>>]\n  Z -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZY<\/SUB><\/FONT>>]\n  X -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>XY<\/SUB><\/FONT>>]\n  }","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<!-- ![](../figs/mvmr_dag.jpg) -->
<p>Here, G are the Genotypes, X are the exposure phenotypes, Y is the
outcome phenotype, and Z are confounders. All except Y are expected to
be multivariate. The edge variables signify effects between these
variables. Let <span class="math inline">\(M\)</span> be the number of
SNPs, <span class="math inline">\(K\)</span> be the number of exposures,
and <span class="math inline">\(J\)</span> be the number of
confounders.</p>
<p>The structural equation model for this DAG is:</p>
<p><span class="math display">\[Z = G\theta_{GZ} + \epsilon_Z\]</span>
<span class="math display">\[X = G\theta_{GX} + Z\theta_{ZX} +
\epsilon_X\]</span> <span class="math display">\[Y = X\theta_{XY} +
G\theta_{GY} + Z\theta_{ZY} + \epsilon_Y\]</span></p>
<p>G is assumed fixed or is drawn from standard normal distributions.
Define</p>
<p><span class="math display">\[\psi_X = \theta_{GZ} *
\theta_{ZX}\]</span> <span class="math display">\[\psi_Y = \theta_{GZ} *
\theta_{ZY}\]</span> Then <span class="math inline">\(\psi_X\)</span>
defines the heritability of X mediated through Z, <span
class="math inline">\(\psi_Y\)</span> defines the confounding effect
from G to Y that is correlated with X, and <span
class="math inline">\(\theta_{GY}\)</span> defines the confounding
effect from G to Y that is not correlated with X.</p>
<p>By default, <span class="math inline">\(\theta_{XY}\)</span> are
fixed effects specified by the user, which allows control over the
strength of effects in the simulation. All other effects <span
class="math inline">\(theta_i\)</span> (where <span
class="math inline">\(i\)</span> is a stand-in for <span
class="math inline">\(GX\)</span>, <span
class="math inline">\(GY\)</span>, and so on) are drawn according to
point-(multivarite-)normal distributions,</p>
<p><span class="math display">\[\theta_i = f_i *
\gamma^*_{i},\]</span></p>
<p>where f is the point-(multivariate-)normal,</p>
<p><span class="math display">\[f_i \sim \pi_{0,i}\delta +
\pi_{1,i}\mathcal{N}_d(\mu, \Sigma_{i}),\]</span></p>
<p>where <span class="math inline">\(\delta\)</span> is the Dirac delta
function and <span class="math inline">\(d\)</span> is the
dimensionality of the effected variable, i.e. <span
class="math inline">\(J\)</span> if the effect is on <span
class="math inline">\(Z\)</span>, <span class="math inline">\(K\)</span>
for <span class="math inline">\(X\)</span>, or <span
class="math inline">\(1\)</span> for <span
class="math inline">\(Y\)</span>. <span
class="math inline">\(\mu\)</span>, the mean parameter, is set to 0 by
default, but can be set to non-zero values to allow “directional
pleiotropy”. <span class="math inline">\(\Sigma_i\)</span> is currently
taken to be a diagonal matrix, but could be generalized to allow
correlated effects.</p>
<p><span class="math inline">\(\pi_{0,i}\)</span> represents the amount
of sparsity while <span class="math inline">\(\pi_{1,i}\)</span>
represents the density, and <span class="math inline">\(\pi_{0,i} +
\pi_{1,i} = 1\)</span>. In practice this is achieved by first simulating
the multivariate normal, then multiplying each entry by <span
class="math inline">\(\pi_{1,i}\)</span>, which is drawn separately for
each entry according to</p>
<p><span class="math display">\[\pi_{1,i} \sim
Bernoulli(\phi_i),\]</span></p>
<p>where <span class="math inline">\(\phi_i\)</span> is a parameter that
controls the level of density. By default, the density of <span
class="math inline">\(\theta_{ZY}\)</span> is set to 1 (<span
class="math inline">\(\phi_{ZY}=1\)</span>) because if some <span
class="math inline">\(Z_j\)</span> does not affect <span
class="math inline">\(Y\)</span> then it is not a confounder.</p>
<p>Finally, <span class="math inline">\(\gamma_i\)</span> represents the
scaling parameter to achieve the desired <span
class="math inline">\(R^2\)</span>. <span
class="math inline">\(G\)</span>, <span
class="math inline">\(Z\)</span>, <span
class="math inline">\(X\)</span>, and <span
class="math inline">\(Y\)</span> are controlled to have unit variance
(see simulation of noise below). For <span
class="math inline">\(G\)</span> to have the desired <span
class="math inline">\(R^2\)</span> (heritability) on <span
class="math inline">\(Z\)</span>, <span
class="math inline">\(X\)</span>, or <span
class="math inline">\(Y\)</span>, we need to adjust this parameter by
the number of SNPs and the sparsity of the effects. Therefore, the
per-SNP <span class="math inline">\(\gamma_i^*\)</span> is</p>
<p><span class="math display">\[\gamma^*_{i} = \sqrt{\gamma_{i} / M /
\phi_i}\]</span></p>
<!-- We draw alpha similarly to theta except that we do not allow sparsity, as a variable that does not affect Y is not a confounder, and a variable that does not affect X does not represent correlated horizontal pleiotropy. However, it may make sense to also allow sparsity for the effects of Z on X, as it could make sense for some Z to affect some subset of X. -->
<p>The noise variances, epsilon, are designed so that Z, X, and Y have
unit variance. So they are simulated according to</p>
<p><span class="math display">\[\epsilon_Z \sim \mathcal{N}_J(0, \xi_Z
I_J)\]</span> <span class="math display">\[\epsilon_X \sim
\mathcal{N}_K(0, \xi_Z I_K)\]</span> <span
class="math display">\[\epsilon_Y \sim \mathcal{N}(0,
\xi_Y)\]</span></p>
<p>where</p>
<p><span class="math display">\[\xi_Z = 1 - \gamma_{GZ}\]</span> <span
class="math display">\[\xi_X = 1 - \gamma_{GX} - \gamma_{ZX}\]</span>
<span class="math display">\[\xi_y = 1 - \gamma_{GY} - \gamma_{ZY} -
\sum_i \theta_{XY,i}^2\]</span></p>
</div>
<div id="simulation-parameters" class="section level2">
<h2>Simulation parameters</h2>
<p>In the initial simulations below, we use the following parameter
settings.</p>
<p>Size parameters:</p>
<ul>
<li><span class="math inline">\(N\)</span> (number of samples) =
10000</li>
<li><span class="math inline">\(K\)</span> (number of exposures) =
30</li>
<li><span class="math inline">\(J\)</span> (number of confounders) =
3</li>
<li>number of GWAS loci = 200</li>
<li><span class="math inline">\(M\)</span> (number of causal SNPs) = up
to 3 per locus; 400 on average</li>
</ul>
<p>Effect size parameters (recall <span
class="math inline">\(\gamma_{GX}\)</span> for example denotes the
effect size of <span class="math inline">\(G\)</span> on <span
class="math inline">\(X\)</span>):</p>
<ul>
<li><span class="math inline">\(\gamma_{GX} = 0.2\)</span></li>
<li><span class="math inline">\(\gamma_{GZ} = 0.4\)</span></li>
<li><span class="math inline">\(\gamma_{GY} = 0.2\)</span> (if
uncorrelated pleiotropy is included)</li>
<li><span class="math inline">\(\gamma_{ZX} = 0.5\)</span></li>
<li><span class="math inline">\(\gamma_{ZY} = 0.375\)</span> (if
correlated pleiotropy is included)</li>
<li><span class="math inline">\(\gamma_{XY} = (\sqrt{0.05}, \sqrt{0.1},
\sqrt{0.2}, \sqrt{0.3})\)</span> for the first four exposures, and 0 for
the others</li>
</ul>
<p>Thus the heritability of exposures is about <span
class="math inline">\(0.2 + 0.4*0.5 = 40\%\)</span> and the heritability
of the outcome is about <span class="math inline">\(0.4*(\sqrt{0.05} +
\sqrt{0.1} + \sqrt{0.2} + \sqrt{0.3}) + 0.4*0.375 = 20.3\%\)</span>
without uncorrelated pleiotropy, or <span
class="math inline">\(40.3\%\)</span> with uncorrelated pleiotropy.</p>
<p>Sparsity parameters (recall <span
class="math inline">\(\phi_{GX}\)</span> for example denotes the density
of effects of <span class="math inline">\(G\)</span> on <span
class="math inline">\(X\)</span>, as in what percentage of the G-X
matrix entries for causal SNPs are nonzero):</p>
<ul>
<li><span class="math inline">\(\phi_{GX} = 0.02\)</span></li>
<li><span class="math inline">\(\phi_{GZ} = 0.1\)</span></li>
<li><span class="math inline">\(\phi_{GY} = 0.02\)</span> (if
uncorrelated pleiotropy is included)</li>
<li><span class="math inline">\(\phi_{ZX} = 0.5\)</span></li>
<li><span class="math inline">\(\phi_{ZY} = 1\)</span> (if correlated
pleiotropy is included)</li>
<li><span class="math inline">\(\phi_{XY} = 4/30\)</span> since four of
the 30 exposures are causal</li>
</ul>
<p>Other info:</p>
<ul>
<li>two-sample mode (exposure and outcome are generated on disjoint
samples, each of size <span class="math inline">\(N/2\)</span>)</li>
<li>results averaged over 100 simulations</li>
</ul>
</div>
<div id="simulation-results-annihilator-mode" class="section level2">
<h2>Simulation results – Annihilator mode</h2>
<p>Here we regress out (“annihilate”) inferred factors from both the
exposure and outcome traits.</p>
<p>Here is a stacked bar plot showing the results versus IVW, with only
correlated pleiotropy. In this plot, factor effects are quite dense
(factors affect 50% of exposures on average) making it relatively easy
for GFA to infer the factors.</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_annih_keep_factors/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr&#39;, &#39;famr_gfa&#39;, &#39;ivw&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/annih_stackbar-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_stackbar-1">
Past versions of annih_stackbar-1.png
</button>
</p>
<div id="fig-annih_stackbar-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_stackbar-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_stackbar-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here is the PIP calibration plot for FAMR_GFA.</p>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_annih_keep_factors/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/famrsims.Rmd/annih_cal-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_cal-1">
Past versions of annih_cal-1.png
</button>
</p>
<div id="fig-annih_cal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_cal-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_cal-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It also does very well at controlling uncorrelated pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_annih_keep_factors/uncor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr&#39;, &#39;famr_gfa&#39;, &#39;ivw&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/annih_stackbar_uncor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_stackbar_uncor-1">
Past versions of annih_stackbar_uncor-1.png
</button>
</p>
<div id="fig-annih_stackbar_uncor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_stackbar_uncor-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_stackbar_uncor-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_annih_keep_factors/uncor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/annih_cal_uncor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_cal_uncor-1">
Past versions of annih_cal_uncor-1.png
</button>
</p>
<div id="fig-annih_cal_uncor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_cal_uncor-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_cal_uncor-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>However, performance degrades a bit if I change factors to only
affect 10% of exposures on average:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_sparser_zx_J3/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr&#39;, &#39;famr_gfa&#39;, &#39;ivw&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/annih_stackbar_sparse-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_stackbar_sparse-1">
Past versions of annih_stackbar_sparse-1.png
</button>
</p>
<div id="fig-annih_stackbar_sparse-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_stackbar_sparse-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_stackbar_sparse-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_sparser_zx_J3/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/annih_cal_sparse-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-annih_cal_sparse-1">
Past versions of annih_cal_sparse-1.png
</button>
</p>
<div id="fig-annih_cal_sparse-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/annih_cal_sparse-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/annih_cal_sparse-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="non-annihilator-mode" class="section level2">
<h2>Non-annihilator mode</h2>
<p>We can also run FAMR just including factors as extra exposures, and
not annihilating / regressing them out. This works a bit worse factor
effects are dense, but roughly as well or even a bit better when factor
effects are sparse.</p>
<p>One odd thing is that, when factors are not annihilated, FAMR works
better when we do not learn SNP “weights” for the Z-scores and R, at
least when we use a highly pruned set of SNPs. We just use the betas as
weights. This is in contrast to annihilator mode, which does better when
we learn weights.</p>
<p>Here are the results with correlated pleiotropy.</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_nolearnwts_noannih_largeish/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr&#39;, &#39;famr_gfa&#39;, &#39;ivw&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/noannih_stackbar-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-noannih_stackbar-1">
Past versions of noannih_stackbar-1.png
</button>
</p>
<div id="fig-noannih_stackbar-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/noannih_stackbar-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/noannih_stackbar-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here is the PIP calibration plot for FAMR_GFA.</p>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_nolearnwts_noannih_largeish/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/noannih_cal-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-noannih_cal-1">
Past versions of noannih_cal-1.png
</button>
</p>
<div id="fig-noannih_cal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/noannih_cal-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/noannih_cal-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="results-with-max-snp-selection" class="section level2">
<h2>Results with Max SNP selection</h2>
<p>Upon further evaluation, it appears that standard MVMR methods like
IVW do better when SNPs are selected according to the “max Z” criterion
rather than just LD clumping.</p>
<p>Here’s what the results look like with correlated pleiotropy, using
FAMR annihilator mode results from above:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_annih_maxsel_spliced/cor_plei_gx_0.2_gy_0.2_gz_0.4_psix_0.2_psiy_0.15_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr&#39;, &#39;famr_gfa&#39;, &#39;ivw&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/maxsel-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-maxsel-1">
Past versions of maxsel-1.png
</button>
</p>
<div id="fig-maxsel-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/maxsel-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/df92d98bde4d3e9970eea6204dcbd4cfde1a9fdc/docs/figure/famrsims.Rmd/maxsel-1.png" target="_blank">df92d98</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="new-non-annihilator-results-with-high-pips-on-factors"
class="section level2">
<h2>New non-annihilator results with high PIPs on factors</h2>
<p>In non-annihilator mode, I have taken a new approach, where I do a
lighter LD prune at abs(R)=0.5, then a stronger LD prune for FA only of
abs(R)=0.1, then using the estimated <span
class="math inline">\(\theta_{ZX}\)</span> from that, I impute <span
class="math inline">\(\theta_{GZ}\)</span> for the rest of the SNPs that
survive the lighter LD pruning threshold. In this case, I learn weights
(with susie_rss) on both exposures and factors.</p>
<p>I also force the prior on factors to be 100x higher than the prior on
the other exposures in order to force susie to fit the factors. This
achieves a sort of balance between the conservatism of annihilator mode
while still retaining the ability (and potential power gains) for
fitting factors alongside exposures.</p>
<p>With correlated pleiotropy only, FAMR-GFA in this setting is somewhat
more powerful than IVW-GFA, but also has a somewhat higher false
positive rate, and is not fully calibrated.</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_gfa_noannih_factor_highpip/cor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_cor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_cor-1">
Past versions of factor_highpip_cor-1.png
</button>
</p>
<div id="fig-factor_highpip_cor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_cor-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_gfa_noannih_factor_highpip/cor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_cor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_cor_calib-1">
Past versions of factor_highpip_cor_calib-1.png
</button>
</p>
<div id="fig-factor_highpip_cor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_cor_calib-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>However, as expected, FAMR does much better with uncorrelated
pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_gfa_noannih_factor_highpip/uncor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_uncor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_uncor-1">
Past versions of factor_highpip_uncor-1.png
</button>
</p>
<div id="fig-factor_highpip_uncor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_uncor-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_gfa_noannih_factor_highpip/uncor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_uncor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_uncor_calib-1">
Past versions of factor_highpip_uncor_calib-1.png
</button>
</p>
<div id="fig-factor_highpip_uncor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_uncor_calib-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Because of that, it does much better overall with both kinds of
pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_gfa_noannih_factor_highpip/both_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_both-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_both-1">
Past versions of factor_highpip_both-1.png
</button>
</p>
<div id="fig-factor_highpip_both-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_both-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_gfa_noannih_factor_highpip/both_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_both_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_both_calib-1">
Past versions of factor_highpip_both_calib-1.png
</button>
</p>
<div id="fig-factor_highpip_both_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_both_calib-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It also has a bit more power than IVW when there is no
pleiotropy.</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_gfa_noannih_factor_highpip/no_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_none-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_none-1">
Past versions of factor_highpip_none-1.png
</button>
</p>
<div id="fig-factor_highpip_none-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_none-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
  &#39;data/results_gfa_noannih_factor_highpip/no_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/factor_highpip_none_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-factor_highpip_none_calib-1">
Past versions of factor_highpip_none_calib-1.png
</button>
</p>
<div id="fig-factor_highpip_none_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/bfabf522a9c046013b15009d5a99f636ade4d03a/docs/figure/famrsims.Rmd/factor_highpip_none_calib-1.png" target="_blank">bfabf52</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="results-with-z-scores-capped-at-20" class="section level2">
<h2>Results with Z-scores capped at 20</h2>
<p>In real data (see ``Metabolite MR’’), sometimes there were issues
with very high Z-scores for exposures computed based on susie weights
when there were many SNPs. Also, the performance of the method in both
simulations and real data tends to be unstable when there are very high
Z-scores. Thus, I tried running some simulations and real data with
exposure z-scores scaled down (equally) so that the maximum z-score
among all exposures was 20.</p>
<p>As shown below, this somewhat reduced performance in simulations.
(This still has high priors on factors, as in the above section.)
However, it did stabilize results in the real data (see ``Metabolite
MR’’).</p>
<p>Here are the results with correlated pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_rescale_z_0pt5impute_finalprune0pt1/cor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/rescale_z_cor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rescale_z_cor-1">
Past versions of rescale_z_cor-1.png
</button>
</p>
<div id="fig-rescale_z_cor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rescale_z_cor-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
 &#39;data/results_rescale_z_0pt5impute_finalprune0pt1/cor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/rescale_z_cor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rescale_z_cor_calib-1">
Past versions of rescale_z_cor_calib-1.png
</button>
</p>
<div id="fig-rescale_z_cor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rescale_z_cor_calib-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And here are the results with uncorrelated pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_rescale_z_0pt5impute_finalprune0pt1/uncor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/rescale_z_uncor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rescale_z_uncor-1">
Past versions of rescale_z_uncor-1.png
</button>
</p>
<div id="fig-rescale_z_uncor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rescale_z_uncor-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
 &#39;data/results_rescale_z_0pt5impute_finalprune0pt1/uncor_plei_gx_0.1_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/rescale_z_uncor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rescale_z_uncor_calib-1">
Past versions of rescale_z_uncor_calib-1.png
</button>
</p>
<div id="fig-rescale_z_uncor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rescale_z_uncor_calib-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="results-with-rsparsepro" class="section level2">
<h2>Results with RSparsePro</h2>
<p><a href="https://doi.org/10.1101/2024.10.29.620968">RSparsePro</a> is
a method that is like Susie-RSS, but is supposed to be more robust to
errors in the R matrix between exposures, factors, and SNPs. Since that
has been a challenge for our method, I decided to try running RSparsePro
in place of Susie-RSS in our simulations. Note that I have not figured
out how RSparsePro can be integrated with a cTWAS-style framework (or if
that is feasible with minimal modification), so right now RSparsePro is
run just once, with no class priors.</p>
<p>This obtained solid results in simulations with no need for strong
priors on factors or rescaling Z-scores, as well as in real data (see
``Metabolite MR’’). However, there are significant computational issues
with the main implementation, which I may be able to resolve by writing
my own version. Also, the PIPs tend to be very dichotomous, i.e. almost
always very close to 0 or 1.</p>
<p>Here are the results with correlated pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;)) </code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_cor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_cor-1">
Past versions of rsparsepro_cor-1.png
</button>
</p>
<div id="fig-rsparsepro_cor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_cor-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
 &#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/cor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;) </code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_cor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_cor_calib-1">
Past versions of rsparsepro_cor_calib-1.png
</button>
</p>
<div id="fig-rsparsepro_cor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_cor_calib-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here are the results with uncorrelated pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/uncor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_uncor-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_uncor-1">
Past versions of rsparsepro_uncor-1.png
</button>
</p>
<div id="fig-rsparsepro_uncor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_uncor-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
 &#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/uncor_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_uncor_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_uncor_calib-1">
Past versions of rsparsepro_uncor_calib-1.png
</button>
</p>
<div id="fig-rsparsepro_uncor_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_uncor_calib-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here are the results with no pleiotropy:</p>
<pre class="r"><code>stacked_barplot(&#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/no_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/&#39;, &#39;1,2,3,4&#39;,
                prefix=&#39;all_res_&#39;, thresh = 0.05, pip_thresh=0.95, use_qval=T,
                methods=c(&#39;famr_gfa&#39;, &#39;ivw_gfa&#39;))</code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_none-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_none-1">
Past versions of rsparsepro_none-1.png
</button>
</p>
<div id="fig-rsparsepro_none-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_none-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res = na.omit(read.table(
 &#39;data/results_test_rsparsepro_final_reg_norescale_prune0pt5/no_plei_gx_0.2_gy_0.2_gz_0.2_psix_0.1_psiy_0.05_mu_0.0/all_res_famr_gfa.txt&#39;))
pip_calibration_plot(res, &#39;1,2,3,4&#39;, &#39;famr_gfa&#39;)</code></pre>
<p><img src="figure/famrsims.Rmd/rsparsepro_none_calib-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rsparsepro_none_calib-1">
Past versions of rsparsepro_none_calib-1.png
</button>
</p>
<div id="fig-rsparsepro_none_calib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6e795457f61f828fe8cbbce89d935a8fae3fb490/docs/figure/famrsims.Rmd/rsparsepro_none_calib-1.png" target="_blank">6e79545</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2024-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_1.1.3     qvalue_2.34.0     ggpubr_0.6.0      ggplot2_3.4.4    
[5] DiagrammeR_1.0.10 workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.4       xfun_0.41          bslib_0.6.1        htmlwidgets_1.6.4 
 [5] visNetwork_2.1.2   processx_3.8.3     rstatix_0.7.2      callr_3.7.3       
 [9] vctrs_0.6.5        tools_4.3.2        ps_1.7.6           generics_0.1.3    
[13] tibble_3.2.1       fansi_1.0.6        highr_0.10         pkgconfig_2.0.3   
[17] RColorBrewer_1.1-3 lifecycle_1.0.4    farver_2.1.1       compiler_4.3.2    
[21] stringr_1.5.1      git2r_0.33.0       munsell_0.5.0      getPass_0.2-4     
[25] carData_3.0-5      httpuv_1.6.13      htmltools_0.5.7    sass_0.4.8        
[29] yaml_2.3.8         crayon_1.5.2       later_1.3.2        pillar_1.9.0      
[33] car_3.1-2          jquerylib_0.1.4    whisker_0.4.1      tidyr_1.3.0       
[37] ellipsis_0.3.2     cachem_1.0.8       abind_1.4-5        tidyselect_1.2.0  
[41] digest_0.6.34      stringi_1.8.3      dplyr_1.1.4        reshape2_1.4.4    
[45] purrr_1.0.2        labeling_0.4.3     splines_4.3.2      rprojroot_2.0.4   
[49] fastmap_1.1.1      grid_4.3.2         colorspace_2.1-0   cli_3.6.2         
[53] magrittr_2.0.3     utf8_1.2.4         broom_1.0.5        withr_3.0.0       
[57] scales_1.3.0       promises_1.2.1     backports_1.4.1    rmarkdown_2.25    
[61] httr_1.4.7         ggsignif_0.6.4     evaluate_0.23      knitr_1.45        
[65] rlang_1.1.3        Rcpp_1.0.12        glue_1.7.0         rstudioapi_0.15.0 
[69] jsonlite_1.8.8     R6_2.5.1           plyr_1.8.9         fs_1.6.3          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
