<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="site_libs/viz-1.8.2/viz.js"></script>
<link href="site_libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="site_libs/grViz-binding-1.0.10/grViz.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mvmr_website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="simulations.html">Simulations</a>
</li>
<li>
  <a href="experimental.html">New Experiments</a>
</li>
<li>
  <a href="realdata.html">Metabolite MR</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nlapier2/mvmr_website">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mvmr_website/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221205code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221205code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221205)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree96b5c8a03cf573a80fddafc2f9703f7dab395295targetblank96b5c8aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nlapier2/mvmr_website/tree/96b5c8a03cf573a80fddafc2f9703f7dab395295" target="_blank">96b5c8a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree96b5c8a03cf573a80fddafc2f9703f7dab395295targetblank96b5c8aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nlapier2/mvmr_website/tree/96b5c8a03cf573a80fddafc2f9703f7dab395295" target="_blank">96b5c8a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    backup/
    Ignored:    data/.DS_Store
    Ignored:    data/results/.DS_Store
    Ignored:    data/results_true_gx/uncor_plei_gy_0.2_zx_0.2_zy_0.2/

Untracked files:
    Untracked:  analysis/realdata.Rmd
    Untracked:  data/real_data_results_summary.tsv

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/b890f047f5314ace4412ce8c2644f7dcbee0cb52/analysis/index.Rmd" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
<td>
rearranged website, added simulation and gibbs tabs, added more project
description
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/b890f047f5314ace4412ce8c2644f7dcbee0cb52/docs/index.html" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
<td>
rearranged website, added simulation and gibbs tabs, added more project
description
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f7f6cbc259f785033b6de4cde214e8d04a37af70/analysis/index.Rmd" target="_blank">f7f6cbc</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
added new results files, including susie results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/f7f6cbc259f785033b6de4cde214e8d04a37af70/docs/index.html" target="_blank">f7f6cbc</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
added new results files, including susie results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/89055eca1732fe21fb35b6b4fea2b0c5d14401af/analysis/index.Rmd" target="_blank">89055ec</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
re-made figures locally in R and added line plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/89055eca1732fe21fb35b6b4fea2b0c5d14401af/docs/index.html" target="_blank">89055ec</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
re-made figures locally in R and added line plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/d2f34a905e9ea1bf5c3bbeb287271a1fc70a2b17/analysis/index.Rmd" target="_blank">d2f34a9</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
remade DAG with diagrammeR, refactored notation, improved description of
methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/d2f34a905e9ea1bf5c3bbeb287271a1fc70a2b17/docs/index.html" target="_blank">d2f34a9</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-28
</td>
<td>
remade DAG with diagrammeR, refactored notation, improved description of
methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/c72e79f643ddfa57ab83bc2edc0fa8f59647d4a1/docs/index.html" target="_blank">c72e79f</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f87ed72cd0a45d63bd8bb277a537bc1ed9061aa8/analysis/index.Rmd" target="_blank">f87ed72</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-05
</td>
<td>
added project overview, simulation description, and simulation results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/05a281bbe03de8adf620c787141102b4739ed475/docs/index.html" target="_blank">05a281b</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/88ab00691df4b9f77b61a1b020331f3b87f3c7d7/analysis/index.Rmd" target="_blank">88ab006</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2022-12-05
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to the MultiVariable Mendelian Randomization (MVMR) project
website. Here we briefly describe the project and provide some
simulation results. In the future we will also include instructions for
running our software.</p>
<p>See <a href="realdata.html">Metabolite MR</a> to see results from the
analysis of real metabolite data, mostly from the UK Biobank and
obtained via the <a href="https://gwas.mrcieu.ac.uk/datasets/">IEU Open
GWAS</a> website.</p>
<div id="project-description" class="section level3">
<h3>Project description</h3>
<div id="htmlwidget-e3d5eff22888eb7ba78a" style="width:800px;height:350px;" class="grViz html-widget "></div>
<script type="application/json" data-for="htmlwidget-e3d5eff22888eb7ba78a">{"x":{"diagram":"digraph {\n  graph [layout = dot, rankdir = LR]\n\n  node [shape = circle]\n  G [label = \"G\"]\n  Z [label = \"Z\"]\n  X [label = \"X\"]\n  Y [label = \"Y\"]\n\n  # edge definitions with the node IDs\n  edge []\n  G -> X [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GX<\/SUB><\/FONT>>]\n  G -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GY<\/SUB><\/FONT>>]\n  G -> Z [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GZ<\/SUB><\/FONT>>]\n  X -> Z [dir=back; label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZX<\/SUB><\/FONT>>]\n  Z -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZY<\/SUB><\/FONT>>]\n  X -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>XY<\/SUB><\/FONT>>]\n  }","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p>In MVMR, we wish to determine which “exposure” phenotype(s) <span
class="math inline">\(X\)</span> affect an “outcome” phenotype Y given
genotypes <span class="math inline">\(G\)</span> that affect <span
class="math inline">\(X\)</span>. <span class="math inline">\(X\)</span>
might be something like metabolites, gene expression, blood cell traits,
etc. <span class="math inline">\(Y\)</span> could be a disease or
another cellular phenotype. An example would be the classic question of
whether HDL, LDL, and/or triglycerides (<span
class="math inline">\(X\)</span>) affect cardiovascular disease (CVD)
risk (<span class="math inline">\(Y\)</span>).</p>
<p>However, if the genetic variants for <span
class="math inline">\(X\)</span> affect <span
class="math inline">\(Y\)</span> through pathways other than <span
class="math inline">\(X\)</span>, this can lead to false positive
conclusions. Suppose <span class="math inline">\(Z\)</span> represents
some gene whose expression level modulates both HDL level and CVD risk.
Then even if HDL doesn’t affect CVD risk, we may still find a false
positive with standard MVMR methods. This is called “confounding”.
Specifically, when <span class="math inline">\(Z\)</span> is a phenotype
caused by <span class="math inline">\(G\)</span> (as opposed to
something like population stratification), this is called “(horizontal)
pleiotropy”.</p>
<p>This model is shown in the “Directed Acyclic Graph (DAG)” above. The
nodes represent the variables and the labels on the edges represent the
effect size of one variable on another. In the DAG above, the pathway
from <span class="math inline">\(G \rightarrow Z \rightarrow Y\)</span>
causes confounding, as does the direct path <span
class="math inline">\(G \rightarrow Y\)</span>, with the effect of this
confounding controlled by the values <span
class="math inline">\(\theta_{GZ} * \theta_{ZY}\)</span> and <span
class="math inline">\(\theta_{GY}\)</span>. The effect <span
class="math inline">\(\theta_{GZ} * \theta_{ZY}\)</span> causes what is
known as “correlated pleiotropy” since it is correlated with the
exposures <span class="math inline">\(X\)</span> (since <span
class="math inline">\(\theta_{GZ}\)</span> affects <span
class="math inline">\(X\)</span>), while the direct effect <span
class="math inline">\(\theta_{GY}\)</span> is called “uncorrelated
pleiotropy” since it is not correlated with <span
class="math inline">\(X\)</span>.</p>
<p>The causal model for the data given this DAG can be represented by a
“Structural Equation Model (SEM)”, which shows how each variable
functionally affects another. The SEM for this DAG is</p>
<p><span class="math display">\[Z = G\theta_{GZ} +
\epsilon_Z\]</span></p>
<p><span class="math display">\[X = G\theta_{GX} + Z\theta_{ZX} +
\epsilon_X\]</span></p>
<p><span class="math display">\[Y = X\theta_{XY} + G\theta_{GY} +
Z\theta_{ZY} + \epsilon_Y\]</span></p>
<p>Our challenge is to estimate <span
class="math inline">\(\theta_{XY}\)</span> despite this confounding. Our
method seeks to address this challenge by modeling potential confounders
and adjusting for this unwanted signal in the data.</p>
</div>
<div id="methods-and-results" class="section level3">
<h3>Methods and Results</h3>
<p>To evaluate existing approaches, I ran some <a
href="simulations.html">simulations</a> on data simulated from the SEM
above. As expected, existing methods had either high False Positive
Rates (FPRs) or very low power due to the confounding.</p>
<p>How do we address this problem? The easiest thing would be if we
already knew the true values of <span
class="math inline">\(\tilde{Z}\)</span> (where <span
class="math inline">\(\tilde{Z} = G\theta_{GZ}\)</span>, ignoring the
noise term) and <span class="math inline">\(\theta_{GY}\)</span>. Then
we could control for the confounding directly. Indeed, my <a
href="simulations.html">simulations</a> show that this “oracle” approach
combined with the standard “two-stage least squares (2SLS)” estimation
procedure controles FPR and maintains high power under arbitrarily
strong confounding.</p>
<p>This is a good sanity check, but in the real world we will not know
<span class="math inline">\(\tilde{Z}\)</span> or <span
class="math inline">\(\theta_{GY}\)</span>. How do we estimate them?
Suppose for now that we have individual-level data available.
<!-- We leave aside estimating $\theta_{GY}$ for now, as this is generally an easier problem -- for example, "MR-Egger" is an existing solution to adjust for uncorrelated pleiotropy. -->
Let’s start by focusing on <span
class="math inline">\(\tilde{Z}\)</span>. In this case, since we already
know <span class="math inline">\(G\)</span>, the challenge is really to
estimate <span class="math inline">\(\theta_{GZ}\)</span>. (An
alternative that also works is to learn <span
class="math inline">\(\psi_X = \theta_{GZ} * \theta_{ZX}\)</span>).</p>
<p>One straightforward approach is to apply truncated singular value
decomposition (SVD), or any other standard factor analysis (FA) method,
to <span class="math inline">\(\tilde{X}\)</span>. From our DAG / SEM,
we have that <span class="math inline">\(\tilde{X} = G * (\theta_{GX} +
\theta_{GZ} * \theta_{ZX})\)</span>. We assume that each <span
class="math inline">\(\tilde{Z}\)</span> affects several exposures. So
our hope is that the latent factors will pick up on these factors
affecting multiple variables.</p>
<p>However, these “one-stage” approaches (FA on <span
class="math inline">\(X\)</span> without accounting for <span
class="math inline">\(Y\)</span>) are unlikely to work satisfactorily.
One reason is that <span class="math inline">\(\theta_{GZ}\)</span>, our
target, is NOT independent of <span class="math inline">\(Y\)</span>,
since <span class="math inline">\(Z\)</span> affects <span
class="math inline">\(Y\)</span> (by definition, since <span
class="math inline">\(Z\)</span> is a confounder). Therefore, not
accounting for <span class="math inline">\(Y\)</span> necessarily limits
how well one can approximate <span
class="math inline">\(\theta_{GZ}\)</span>.</p>
<p>My <a href="gibbs.html">Gibbs sampler page</a> shows the conditional
distribution of <span class="math inline">\(\theta_{GZ}\)</span> given
all other variables, assuming that they are all normally-distributed.
This explicitly shows how <span
class="math inline">\(\theta_{GZ}\)</span> depends on both X and Y.
However, there are currently some limitations for the multiple-exposure
case that I am working through, and I am not currently accounting for
sparsity or non-normal unimodal distributions, like ash does.</p>
<p>For summary statistics, where <span class="math inline">\(G\)</span>
is not available, we will probably have to perform FA on the betas and
stderrs for <span class="math inline">\(X\)</span> and <span
class="math inline">\(Y\)</span>. One starting point is to assume that
the direct effects of any given SNP on the exposures are independent, so
that the betas for each exposure are only correlated due to the
unobserved factors. We could explore extensions to existing Bayesian
frameworks like RUV-Bayes, Flash, or Mash.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DiagrammeR_1.0.10 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12        RColorBrewer_1.1-3 compiler_4.2.1     pillar_1.9.0      
 [5] bslib_0.5.0        later_1.3.1        git2r_0.32.0       jquerylib_0.1.4   
 [9] tools_4.2.1        getPass_0.2-2      digest_0.6.33      jsonlite_1.8.7    
[13] evaluate_0.21      lifecycle_1.0.4    tibble_3.2.1       pkgconfig_2.0.3   
[17] rlang_1.1.3        cli_3.6.2          rstudioapi_0.15.0  yaml_2.3.7        
[21] xfun_0.39          fastmap_1.1.1      httr_1.4.6         stringr_1.5.0     
[25] knitr_1.43         htmlwidgets_1.6.2  sass_0.4.7         fs_1.6.3          
[29] vctrs_0.6.5        rprojroot_2.0.3    glue_1.7.0         R6_2.5.1          
[33] processx_3.8.2     fansi_1.0.6        rmarkdown_2.23     callr_3.7.3       
[37] magrittr_2.0.3     whisker_0.4.1      ellipsis_0.3.2     ps_1.7.5          
[41] promises_1.2.0.1   htmltools_0.5.5    httpuv_1.6.11      utf8_1.2.4        
[45] stringi_1.8.3      visNetwork_2.1.2   cachem_1.0.8      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
