<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Simulations</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="site_libs/viz-1.8.2/viz.js"></script>
<link href="site_libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="site_libs/grViz-binding-1.0.10/grViz.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mvmr_website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="simulations.html">Simulations</a>
</li>
<li>
  <a href="experimental.html">New Experiments</a>
</li>
<li>
  <a href="realdata.html">Metabolite MR</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nlapier2/mvmr_website">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulations</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>mvmr_website/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221205code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221205)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221205code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221205)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree96b5c8a03cf573a80fddafc2f9703f7dab395295targetblank96b5c8aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/nlapier2/mvmr_website/tree/96b5c8a03cf573a80fddafc2f9703f7dab395295" target="_blank">96b5c8a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomnlapier2mvmrwebsitetree96b5c8a03cf573a80fddafc2f9703f7dab395295targetblank96b5c8aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/nlapier2/mvmr_website/tree/96b5c8a03cf573a80fddafc2f9703f7dab395295" target="_blank">96b5c8a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    backup/
    Ignored:    data/.DS_Store
    Ignored:    data/results/.DS_Store
    Ignored:    data/results_true_gx/uncor_plei_gy_0.2_zx_0.2_zy_0.2/

Untracked files:
    Untracked:  analysis/realdata.Rmd
    Untracked:  data/real_data_results_summary.tsv

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/experimental.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/simulations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/simulations.Rmd</code>) and HTML
(<code>docs/simulations.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/analysis/simulations.Rmd" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
<td>
added brms, susie.ss, and mrash.ss results; split into new analysis
file, experimental.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/simulations.html" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
<td>
added brms, susie.ss, and mrash.ss results; split into new analysis
file, experimental.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/a1dd33895805194a34a0d3a45573b275c5e9daac/analysis/simulations.Rmd" target="_blank">a1dd338</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-09-05
</td>
<td>
added true G-X sim and preliminary real data results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/a1dd33895805194a34a0d3a45573b275c5e9daac/docs/simulations.html" target="_blank">a1dd338</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-09-05
</td>
<td>
added true G-X sim and preliminary real data results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/analysis/simulations.Rmd" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
<td>
added high-exposure, uk biobank, and snp selection results to website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/simulations.html" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
<td>
added high-exposure, uk biobank, and snp selection results to website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/analysis/simulations.Rmd" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
<td>
added mrash and ps sims
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/08e3dae8978a7721593da93237f214f94ff76f27/docs/simulations.html" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
<td>
added mrash and ps sims
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f5f65b62913292e68d0b5a3f2dba26dc1f80b36c/analysis/simulations.Rmd" target="_blank">f5f65b6</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
<td>
updated check for pip methods in plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/analysis/simulations.Rmd" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
<td>
added in GFA sims and removed old RPCA and "new parameter setting" sims
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/simulations.html" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
<td>
added in GFA sims and removed old RPCA and "new parameter setting" sims
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/analysis/simulations.Rmd" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
<td>
added rpca-cate augmented experiments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/docs/simulations.html" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
<td>
added rpca-cate augmented experiments
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/729493ac80bb739387a0fa3ccbe2fd4f1ea40a61/analysis/simulations.Rmd" target="_blank">729493a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-28
</td>
<td>
added fixed highsnp sims and removed old ones
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/729493ac80bb739387a0fa3ccbe2fd4f1ea40a61/docs/simulations.html" target="_blank">729493a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-28
</td>
<td>
added fixed highsnp sims and removed old ones
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f8ef21640c3d74e63771218a4ca3159e8413fde0/analysis/simulations.Rmd" target="_blank">f8ef216</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-22
</td>
<td>
added cor only and uncor only plots to original sim part
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/f8ef21640c3d74e63771218a4ca3159e8413fde0/docs/simulations.html" target="_blank">f8ef216</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-22
</td>
<td>
added cor only and uncor only plots to original sim part
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/543a43982fc7cd7aafeb810beec66f3deee78ebb/analysis/simulations.Rmd" target="_blank">543a439</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-21
</td>
<td>
added susiecate plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/543a43982fc7cd7aafeb810beec66f3deee78ebb/docs/simulations.html" target="_blank">543a439</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-21
</td>
<td>
added susiecate plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/d4deb61cd109b235916c50d2e78ae2881507bfc8/analysis/simulations.Rmd" target="_blank">d4deb61</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-09
</td>
<td>
added preliminary 2sls-cate results in correlated-only case
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/d4deb61cd109b235916c50d2e78ae2881507bfc8/docs/simulations.html" target="_blank">d4deb61</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-09
</td>
<td>
added preliminary 2sls-cate results in correlated-only case
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f926bb8d91429490450910f87d0974b953af0592/analysis/simulations.Rmd" target="_blank">f926bb8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-01
</td>
<td>
added simulation plots showing power at given FPR threshold
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/f926bb8d91429490450910f87d0974b953af0592/docs/simulations.html" target="_blank">f926bb8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-01
</td>
<td>
added simulation plots showing power at given FPR threshold
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e8b7ae94e157e72b5d166dcf8047011c6bfb5b71/analysis/simulations.Rmd" target="_blank">e8b7ae9</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-01
</td>
<td>
added 2SLS Enhanced results on original data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/e8b7ae94e157e72b5d166dcf8047011c6bfb5b71/docs/simulations.html" target="_blank">e8b7ae9</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-01
</td>
<td>
added 2SLS Enhanced results on original data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f249398a26e89e3fb4c60adedb3f9492751372c2/analysis/simulations.Rmd" target="_blank">f249398</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-18
</td>
<td>
added some info on gibbs results, updated sims page to get updated
navbar
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/f249398a26e89e3fb4c60adedb3f9492751372c2/docs/simulations.html" target="_blank">f249398</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-18
</td>
<td>
added some info on gibbs results, updated sims page to get updated
navbar
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/b890f047f5314ace4412ce8c2644f7dcbee0cb52/analysis/simulations.Rmd" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
<td>
rearranged website, added simulation and gibbs tabs, added more project
description
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/nlapier2/mvmr_website/b890f047f5314ace4412ce8c2644f7dcbee0cb52/docs/simulations.html" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
<td>
rearranged website, added simulation and gibbs tabs, added more project
description
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="simulation-description" class="section level2">
<h2>Simulation description</h2>
<p>We simulate according to the following DAG:</p>
<div id="htmlwidget-e3d5eff22888eb7ba78a" style="width:800px;height:350px;" class="grViz html-widget "></div>
<script type="application/json" data-for="htmlwidget-e3d5eff22888eb7ba78a">{"x":{"diagram":"digraph {\n  graph [layout = dot, rankdir = LR]\n\n  node [shape = circle]\n  G [label = \"G\"]\n  Z [label = \"Z\"]\n  X [label = \"X\"]\n  Y [label = \"Y\"]\n\n  # edge definitions with the node IDs\n  edge []\n  G -> X [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GX<\/SUB><\/FONT>>]\n  G -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GY<\/SUB><\/FONT>>]\n  G -> Z [label=<θ<FONT POINT-SIZE=\"8\"><SUB>GZ<\/SUB><\/FONT>>]\n  X -> Z [dir=back; label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZX<\/SUB><\/FONT>>]\n  Z -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>ZY<\/SUB><\/FONT>>]\n  X -> Y [label=<θ<FONT POINT-SIZE=\"8\"><SUB>XY<\/SUB><\/FONT>>]\n  }","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<!-- ![](../figs/mvmr_dag.jpg) -->
<p>Here, G are the Genotypes, X are the exposure phenotypes, Y is the
outcome phenotype, and Z are confounders. All except Y are expected to
be multivariate. The edge variables signify effects between these
variables. Let <span class="math inline">\(M\)</span> be the number of
SNPs, <span class="math inline">\(K\)</span> be the number of exposures,
and <span class="math inline">\(J\)</span> be the number of
confounders.</p>
<p>The structural equation model for this DAG is:</p>
<p><span class="math display">\[Z = G\theta_{GZ} + \epsilon_Z\]</span>
<span class="math display">\[X = G\theta_{GX} + Z\theta_{ZX} +
\epsilon_X\]</span> <span class="math display">\[Y = X\theta_{XY} +
G\theta_{GY} + Z\theta_{ZY} + \epsilon_Y\]</span></p>
<p>G is assumed fixed or is drawn from standard normal distributions.
Define</p>
<p><span class="math display">\[\psi_X = \theta_{GZ} *
\theta_{ZX}\]</span> <span class="math display">\[\psi_Y = \theta_{GZ} *
\theta_{ZY}\]</span> Then <span class="math inline">\(\psi_X\)</span>
defines the heritability of X mediated through Z, <span
class="math inline">\(\psi_Y\)</span> defines the confounding effect
from G to Y that is correlated with X, and <span
class="math inline">\(\theta_{GY}\)</span> defines the confounding
effect from G to Y that is not correlated with X.</p>
<p>By default, <span class="math inline">\(\theta_{XY}\)</span> are
fixed effects specified by the user, which allows control over the
strength of effects in the simulation. All other effects <span
class="math inline">\(theta_i\)</span> (where <span
class="math inline">\(i\)</span> is a stand-in for <span
class="math inline">\(GX\)</span>, <span
class="math inline">\(GY\)</span>, and so on) are drawn according to
point-(multivarite-)normal distributions,</p>
<p><span class="math display">\[\theta_i = f_i *
\gamma^*_{i},\]</span></p>
<p>where f is the point-(multivariate-)normal,</p>
<p><span class="math display">\[f_i \sim \pi_{0,i}\delta +
\pi_{1,i}\mathcal{N}_d(\mu, \Sigma_{i}),\]</span></p>
<p>where <span class="math inline">\(\delta\)</span> is the Dirac delta
function and <span class="math inline">\(d\)</span> is the
dimensionality of the effected variable, i.e. <span
class="math inline">\(J\)</span> if the effect is on <span
class="math inline">\(Z\)</span>, <span class="math inline">\(K\)</span>
for <span class="math inline">\(X\)</span>, or <span
class="math inline">\(1\)</span> for <span
class="math inline">\(Y\)</span>. <span
class="math inline">\(\mu\)</span>, the mean parameter, is set to 0 by
default, but can be set to non-zero values to allow “directional
pleiotropy”. <span class="math inline">\(\Sigma_i\)</span> is currently
taken to be a diagonal matrix, but could be generalized to allow
correlated effects.</p>
<p><span class="math inline">\(\pi_{0,i}\)</span> represents the amount
of sparsity while <span class="math inline">\(\pi_{1,i}\)</span>
represents the density, and <span class="math inline">\(\pi_{0,i} +
\pi_{1,i} = 1\)</span>. In practice this is achieved by first simulating
the multivariate normal, then multiplying each entry by <span
class="math inline">\(\pi_{1,i}\)</span>, which is drawn separately for
each entry according to</p>
<p><span class="math display">\[\pi_{1,i} \sim
Bernoulli(\phi_i),\]</span></p>
<p>where <span class="math inline">\(\phi_i\)</span> is a parameter that
controls the level of density. By default, the density of <span
class="math inline">\(\theta_{ZY}\)</span> is set to 1 (<span
class="math inline">\(\phi_{ZY}=1\)</span>) because if some <span
class="math inline">\(Z_j\)</span> does not affect <span
class="math inline">\(Y\)</span> then it is not a confounder.</p>
<p>Finally, <span class="math inline">\(\gamma_i\)</span> represents the
scaling parameter to achieve the desired <span
class="math inline">\(R^2\)</span>. <span
class="math inline">\(G\)</span>, <span
class="math inline">\(Z\)</span>, <span
class="math inline">\(X\)</span>, and <span
class="math inline">\(Y\)</span> are controlled to have unit variance
(see simulation of noise below). For <span
class="math inline">\(G\)</span> to have the desired <span
class="math inline">\(R^2\)</span> (heritability) on <span
class="math inline">\(Z\)</span>, <span
class="math inline">\(X\)</span>, or <span
class="math inline">\(Y\)</span>, we need to adjust this parameter by
the number of SNPs and the sparsity of the effects. Therefore, the
per-SNP <span class="math inline">\(\gamma_i^*\)</span> is</p>
<p><span class="math display">\[\gamma^*_{i} = \sqrt{\gamma_{i} / M /
\phi_i}\]</span></p>
<!-- We draw alpha similarly to theta except that we do not allow sparsity, as a variable that does not affect Y is not a confounder, and a variable that does not affect X does not represent correlated horizontal pleiotropy. However, it may make sense to also allow sparsity for the effects of Z on X, as it could make sense for some Z to affect some subset of X. -->
<p>The noise variances, epsilon, are designed so that Z, X, and Y have
unit variance. So they are simulated according to</p>
<p><span class="math display">\[\epsilon_Z \sim \mathcal{N}_J(0, \xi_Z
I_J)\]</span> <span class="math display">\[\epsilon_X \sim
\mathcal{N}_K(0, \xi_Z I_K)\]</span> <span
class="math display">\[\epsilon_Y \sim \mathcal{N}(0,
\xi_Y)\]</span></p>
<p>where</p>
<p><span class="math display">\[\xi_Z = 1 - \gamma_{GZ}\]</span> <span
class="math display">\[\xi_X = 1 - \gamma_{GX} - \gamma_{ZX}\]</span>
<span class="math display">\[\xi_y = 1 - \gamma_{GY} - \gamma_{ZY} -
\sum_i \theta_{XY,i}^2\]</span></p>
<div id="baseline-methods-assessed" class="section level3">
<h3>Baseline Methods Assessed</h3>
<ul>
<li>“2SLS”: vanilla two-stage least squares implemented by me. I first
perform multiple regression of each <span
class="math inline">\(X_j\)</span> on <span
class="math inline">\(G\)</span>, obtaining an estimate <span
class="math inline">\(\hat{\tilde{X}}\)</span> of the genetic component
of X, <span class="math inline">\(\tilde{X}\)</span>. I then regress
<span class="math inline">\(Y\)</span> on <span
class="math inline">\(\hat{\tilde{X}}\)</span> (second-stage
regression).</li>
</ul>
<p><span class="math display">\[X_1,...,X_K \sim G \rightarrow
\hat{\tilde{X}}\]</span> <span class="math display">\[Y \sim
\hat{\tilde{X}} \rightarrow \hat{\theta}_{XY}, pvalue\]</span></p>
<ul>
<li>“2SLS Oracle”: This is an augmented version of 2SLS where, in the
second-stage regression, I include the true values of <span
class="math inline">\(G\theta_{GZ}\)</span> (which is equal to the
genetic component of Z, <span class="math inline">\(\tilde{Z}\)</span>)
and <span class="math inline">\(G\theta_{GY}\)</span> as covariates.
This represents the ideal performance of 2SLS if the true confounders
were known (“oracle” setting), though it may be beatable by putting
appropriate sparse priors on the confounders.</li>
</ul>
<p><span class="math display">\[X_1,...,X_K \sim G \rightarrow
\hat{\tilde{X}}\]</span> <span class="math display">\[Y \sim
\hat{\tilde{X}} + \tilde{Z} + G\theta_{GY} \rightarrow
\hat{\theta}_{XY}, pvalue\]</span></p>
<ul>
<li>“2SLS Enhanced”: This is an augmented version of 2SLS where, in the
second stage regression, I include estimates of <span
class="math inline">\(G\theta_{GZ}\)</span> and <span
class="math inline">\(G\theta_{GY}\)</span> that can be obtained without
knowing the true values. The former is estimated by running SVD on <span
class="math inline">\(\hat{\theta}_{GX}\)</span>. The latter is
estimated by regressing <span class="math inline">\(X\)</span> out of
<span class="math inline">\(Y\)</span> and then regressing those
residuals on <span class="math inline">\(G\)</span>.</li>
</ul>
<p><span class="math display">\[X_1,...,X_K \sim G \rightarrow
\hat{\tilde{X}}\]</span></p>
<p><span class="math display">\[SVD(\hat{\theta}_{GX}) \rightarrow
\widehat{\theta_{GZ}\theta_{ZX}} = \hat{\psi}_X\]</span> <span
class="math display">\[Y \sim X \rightarrow \hat{Y}_{resid}\]</span>
<span class="math display">\[\hat{Y}_{resid} \sim G \rightarrow
\hat{\theta}_{GY}\]</span> <span class="math display">\[Y \sim
\hat{\tilde{X}} + G\hat{\psi}_x + G\hat{\theta}_{GY} \rightarrow
\hat{\theta}_{XY}, pvalue\]</span></p>
<ul>
<li>SuSiE-SVD: This performs the same first-stage regression as 2SLS,
then runs runs the SuSiE R package for the second-stage regression.
Along with <span class="math inline">\(\hat{\tilde{X}}\)</span>, we
include the SNPs <span class="math inline">\(G\)</span> to attempt to
control for uncorrelated pleiotropy as well as an estimate of the
confounders <span class="math inline">\(\hat{\tilde{Z}}\)</span>,
produced by running truncated singular value decomposition (SVD) on
<span class="math inline">\(\hat{\tilde{X}}\)</span>, to attempt to
control for correlated pleiotropy. We give a slightly incorrect value
for the true number of confounders to SVD.</li>
</ul>
<p><span class="math display">\[X_1,...,X_K \sim G \rightarrow
\hat{\tilde{X}}\]</span> <span
class="math display">\[SVD(\hat{\tilde{X}}) \rightarrow \hat{\tilde{Z}}
= G\hat{\theta}_{GZ}\]</span></p>
<p><span class="math display">\[susieR::susie(Y \sim \hat{\tilde{X}} + G
+ \hat{\tilde{Z}}) \rightarrow \hat{\theta}_{XY}, pvalue\]</span></p>
<ul>
<li>Multivariate versions of four standard MR methods, implemented in
the MendelianRandomization package: IVW, Egger, Median, and Lasso. For
each of these, I first regress each <span
class="math inline">\(X_j\)</span> and <span
class="math inline">\(Y\)</span> on each genetic variant. The betas and
standard errors from these per-variant regressions (mimicking summary
statistics) are given to the methods as input.</li>
</ul>
<p><span class="math display">\[X_i \sim G_j \rightarrow
\hat{\beta}_{ij}, \hat{s}^2_{ij} \qquad Y \sim G_j \rightarrow
\hat{\beta}_{yj}, \hat{s}^2_{yj}\]</span> <span
class="math display">\[MendelianRandomization::MVMR\_IVW(\hat{\beta},
\hat{s}^2) \rightarrow \hat{\theta}_{XY}, pvalue\]</span></p>
</div>
</div>
<div id="simulation-results" class="section level2">
<h2>Simulation Results</h2>
<p>Here are some plots showing our simulation results. First, I’ll
briefly explain the parameter settings.</p>
<p>The following parameters are fixed:</p>
<ul>
<li>N = 20000 samples / individuals</li>
<li>M = 100 SNPs</li>
<li>K = 30 exposures / risk factors</li>
<li>First four exposures are true effects with <span
class="math inline">\(\theta_{XY}\)</span> = 0.05 / 0.1 / 0.2 / 0.3.
This means that the variance explained of the outcome is 0.0025, 0.01,
0.04, 0.09. The other 26 effects are null. I show the first four nulls
in the bar plots below for illustrative purposes.</li>
<li>J = 3 correlated confounding variables “Z”</li>
<li>50% sparsity, i.e. of genotype effects on X/Y/Z are set to zero. The
rest are drawn from normal distributions. In other words, the effects
are point-normal. See “About” for details.</li>
</ul>
<p>There are five key parameters I vary in these simulations:</p>
<ul>
<li><span class="math inline">\(\gamma_{GZ}\)</span>/<span
class="math inline">\(\gamma_{GY}\)</span>/<span
class="math inline">\(\gamma_{GZ}\)</span> = variance of X/Y/Z explained
by G;</li>
<li><span class="math inline">\(\psi_{X}\)</span>/<span
class="math inline">\(\psi_{Y}\)</span> = percent of X/Y explained by G
through confounder Z;</li>
<li><span class="math inline">\(\mu\)</span> = mean parameter of
multivariate normal effect size draws</li>
</ul>
<p>The default settings of these are 0.1 (10%) for <span
class="math inline">\(\gamma_{GZ}\)</span>/<span
class="math inline">\(\gamma_{GY}\)</span>/<span
class="math inline">\(\psi_{X}\)</span>/<span
class="math inline">\(\psi_{Y}\)</span> and 0 for <span
class="math inline">\(\mu\)</span>. The default setting for <span
class="math inline">\(\gamma_{GZ}\)</span> is 0.6, which is high, but
this is to allow a wide range of <span
class="math inline">\(\psi_{X}\)</span> and <span
class="math inline">\(\psi_{Y}\)</span> values, since <span
class="math inline">\(\psi_{X}\)</span> and <span
class="math inline">\(\psi_{Y}\)</span> cannot be greater than <span
class="math inline">\(\gamma_{GZ}\)</span>.</p>
<p>I show two types of plots:</p>
<ul>
<li>Bar plots, showing results for a single parameter setting, with
Positive Rate (FPR/Power) on the y-axis and the index of the exposure on
the x-axis. The first four exposures are true positives (so y-axis is
empirical power) while the rest are null (so y-axis is FPR); only the
first four nulls are shown for brevity.</li>
<li>Line plots, showing results as one of the parameters is varied. FPR
plots show results averaged over the four true effects while Power plots
show results averaged over the null effects. FPR/Power is on the y-axis
and the parameter value is on the x-axis.</li>
</ul>
<p>Here are the results with the default settings. We see that all
methods except 2SLS-Oracle are inflated, with 2SLS-Oracle being the most
powerful. Apart from 2SLS-Oracle, there is a clear order among the other
methods in terms of both FPR and Power, with 2SLS &gt; MVMR Lasso &gt;
MVMR Median &gt; MVMR IVW/Egger.</p>
<p><img src="figure/simulations.Rmd/both_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-both_plei-1">
Past versions of both_plei-1.png
</button>
</p>
<div id="fig-both_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/docs/figure/simulations.Rmd/both_plei-1.png" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/b890f047f5314ace4412ce8c2644f7dcbee0cb52/docs/figure/simulations.Rmd/both_plei-1.png" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As a sanity check, the plot below shows results in simulations with
no confounding. As expected, all methods perform well.</p>
<p><img src="figure/simulations.Rmd/no_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no_plei-1">
Past versions of no_plei-1.png
</button>
</p>
<div id="fig-no_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/docs/figure/simulations.Rmd/no_plei-1.png" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/b890f047f5314ace4412ce8c2644f7dcbee0cb52/docs/figure/simulations.Rmd/no_plei-1.png" target="_blank">b890f04</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-01-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here’s the performance with only correlated pleiotropy. Here we see
that all methods except the oracle method have inflated FPR:</p>
<p><img src="figure/simulations.Rmd/cor_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei-1">
Past versions of cor_plei-1.png
</button>
</p>
<div id="fig-cor_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/docs/figure/simulations.Rmd/cor_plei-1.png" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f8ef21640c3d74e63771218a4ca3159e8413fde0/docs/figure/simulations.Rmd/cor_plei-1.png" target="_blank">f8ef216</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, here’s the performance with only uncorrelated
pleiotropy.</p>
<p><img src="figure/simulations.Rmd/uncor_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-uncor_plei-1">
Past versions of uncor_plei-1.png
</button>
</p>
<div id="fig-uncor_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/c9438a851bbbd7d4e2d4773be3f38ad275f45fdd/docs/figure/simulations.Rmd/uncor_plei-1.png" target="_blank">c9438a8</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-04-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/f8ef21640c3d74e63771218a4ca3159e8413fde0/docs/figure/simulations.Rmd/uncor_plei-1.png" target="_blank">f8ef216</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="simulations-with-new-settings-cate-rpca-gfa-plugins"
class="section level2">
<h2>Simulations with new settings + Cate, RPCA, &amp; GFA plugins</h2>
<p>I performed some more simulations under settings that may be more
realistic. The following changes were made as compared with the above
simulations:</p>
<!-- * The number of samples was increased from 20,000 to 50,000 -->
<!-- * As a result of the parameter changes, some of the values they varied between were changed. -->
<ul>
<li>The number of SNPs was increased from 100 to 300</li>
<li>The density parameters were decreased from <span
class="math inline">\(\phi_X=\phi_Y=\phi_Z=50\%\)</span> to <span
class="math inline">\(\phi_X=10\%, \phi_Y=\phi_Z=10\%\)</span>; thus a
much smaller percentage of variants will affect the phenotypes,
especially for pleiotropic effects.</li>
<li>The defaults for <span class="math inline">\(\theta_{GX}\)</span>
and <span class="math inline">\(\theta_{GY}\)</span> were increased from
<span class="math inline">\(0.1\)</span> to <span
class="math inline">\(0.2\)</span>. Meanwhile the default for <span
class="math inline">\(\theta_{GZ}\)</span> was decreased to <span
class="math inline">\(0.4\)</span> and <span
class="math inline">\(\psi_Y\)</span> was kept at <span
class="math inline">\(0.15\)</span>. This was done to obtain more
balanced and realistic heritabilities, something like we might expect if
<span class="math inline">\(X\)</span> were lipid measurements and <span
class="math inline">\(Y\)</span> was some complex trait. The resulting
heritabilities are <span class="math inline">\(h^2(X)=30\%\)</span>,
<span class="math inline">\(h^2(Z)=40\%\)</span>, <span
class="math inline">\(h^2(Y)=49.25\%\)</span>.</li>
<li>I didn’t bother fiddling with <span
class="math inline">\(\theta_{GX}\)</span> or <span
class="math inline">\(\mu\)</span> here as the effects should be
similar.</li>
<li>I now include more methods in the plots.</li>
</ul>
<p>Here I’m comparing standard methods augmented with either Cate or
RPCA, under several sparsity and SNP count settings. (Update: I have now
added GFA as well, which also works on summary statistics.) This will
show that Cate is more robust than RPCA in that it does not assume that
the G–&gt;Z effects are dense, while RPCA needs this assumption. Of
course, the benefit of RPCA is we can apply it to summary statistics,
while Cate needs individual-level data.</p>
<p>We will also see that Robust generally enjoys greater power than
SuSiE, but is occasionally inflated, particularly in a scenario with
denser direct effects and fewer SNPs overall. On the other hand, SuSiE
has low power to detect weak effects but is very robust to almost any
setting, provided that it is given a reasonable estimate of correlated
pleiotropic effects via Cate or RPCA.</p>
<p>Here are the results with correlated pleiotropy only, with <span
class="math inline">\(\theta_{gx}\)</span> sparse and <span
class="math inline">\(\theta_{gz}\)</span> dense (<span
class="math inline">\(\phi_{gx}=0.1\)</span> and <span
class="math inline">\(\phi_{gz}=0.8\)</span>).</p>
<p><img src="figure/simulations.Rmd/cor_plei_gfa1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_gfa1-1">
Past versions of cor_plei_gfa1-1.png
</button>
</p>
<div id="fig-cor_plei_gfa1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_gfa1-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/figure/simulations.Rmd/cor_plei_gfa1-1.png" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here’s the same plot, except with <span
class="math inline">\(\phi_{gz}=0.1\)</span>. This is a problem for
RPCA, but not Cate.</p>
<p><img src="figure/simulations.Rmd/cor_plei_gfa2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_gfa2-1">
Past versions of cor_plei_gfa2-1.png
</button>
</p>
<div id="fig-cor_plei_gfa2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_gfa2-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/figure/simulations.Rmd/cor_plei_gfa2-1.png" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, here’s a setting with <span
class="math inline">\(100\)</span> SNPs (the above two have 300) and
<span class="math inline">\(\phi_{gx}=0.5\)</span> and <span
class="math inline">\(\phi_{gy}=0.4\)</span>.</p>
<p><img src="figure/simulations.Rmd/cor_plei_gfa4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_gfa4-1">
Past versions of cor_plei_gfa4-1.png
</button>
</p>
<div id="fig-cor_plei_gfa4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_gfa4-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/figure/simulations.Rmd/cor_plei_gfa4-1.png" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here’s the same, but with both kinds of pleiotropy:</p>
<p><img src="figure/simulations.Rmd/both_plei_gfa4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-both_plei_gfa4-1">
Past versions of both_plei_gfa4-1.png
</button>
</p>
<div id="fig-both_plei_gfa4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/both_plei_gfa4-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/figure/simulations.Rmd/both_plei_gfa4-1.png" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, here’s the power at a fixed FPR threshold, using the same
data as above:</p>
<p><img src="figure/simulations.Rmd/powfpr_both_plei_gfa4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-powfpr_both_plei_gfa4-1">
Past versions of powfpr_both_plei_gfa4-1.png
</button>
</p>
<div id="fig-powfpr_both_plei_gfa4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/e2a2fc4e2989f1f53e2598b63d2b389101b73765/docs/figure/simulations.Rmd/powfpr_both_plei_gfa4-1.png" target="_blank">e2a2fc4</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mr.ash-assessments" class="section level2">
<h2>MR.Ash assessments</h2>
<p>Last four plots above, but now comparing MR.Ash to other second-stage
methods.</p>
<p>Correlated pleiotropy, <span
class="math inline">\(\phi_{gy}=0.1\)</span>.</p>
<p><img src="figure/simulations.Rmd/cor_plei_mrash2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_mrash2-1">
Past versions of cor_plei_mrash2-1.png
</button>
</p>
<div id="fig-cor_plei_mrash2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_mrash2-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/cor_plei_mrash2-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Correlated pleiotropy, with <span class="math inline">\(100\)</span>
SNPs (the above has 300) and <span
class="math inline">\(\phi_{gx}=0.5\)</span> and <span
class="math inline">\(\phi_{gy}=0.4\)</span>.</p>
<p><img src="figure/simulations.Rmd/cor_plei_mrash4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_mrash4-1">
Past versions of cor_plei_mrash4-1.png
</button>
</p>
<div id="fig-cor_plei_mrash4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_mrash4-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/cor_plei_mrash4-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here’s the same, but with both kinds of pleiotropy:</p>
<p><img src="figure/simulations.Rmd/both_plei_mrash4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-both_plei_mrash4-1">
Past versions of both_plei_mrash4-1.png
</button>
</p>
<div id="fig-both_plei_mrash4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/both_plei_mrash4-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/both_plei_mrash4-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, here’s the power at a fixed FPR threshold, using the same
data as above:</p>
<p><img src="figure/simulations.Rmd/powfpr_both_plei_mrash4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-powfpr_both_plei_mrash4-1">
Past versions of powfpr_both_plei_mrash4-1.png
</button>
</p>
<div id="fig-powfpr_both_plei_mrash4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/powfpr_both_plei_mrash4-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="simulations-with-many-exposures" class="section level2">
<h2>Simulations with many exposures</h2>
<p>We have talked about potentially supporting hundreds of exposures,
but thus far our simulations have only had 30. Here I raise that to 200.
First, we keep the other settings the same as above, with 300 SNPs.
Notably, there’s still only four true causal exposures, so very sparse,
and also only three confounders. Here’s what it looks like with no
pleiotropy:</p>
<p><img src="figure/simulations.Rmd/200exp_300snp_no_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-200exp_300snp_no_plei-1">
Past versions of 200exp_300snp_no_plei-1.png
</button>
</p>
<div id="fig-200exp_300snp_no_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/200exp_300snp_no_plei-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/200exp_300snp_no_plei-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Right off the bat we notice something odd: robust is inflated despite
no pleiotropy. Everything else looks normal right now.</p>
<p>If we instead increase the number of SNPs to 1000, that goes
away:</p>
<p><img src="figure/simulations.Rmd/200exp_1000snp_no_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-200exp_1000snp_no_plei-1">
Past versions of 200exp_1000snp_no_plei-1.png
</button>
</p>
<div id="fig-200exp_1000snp_no_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/200exp_1000snp_no_plei-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/200exp_1000snp_no_plei-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now here’s correlated-only pleiotropy:</p>
<p><img src="figure/simulations.Rmd/200exp_1000snp_cor_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-200exp_1000snp_cor_plei-1">
Past versions of 200exp_1000snp_cor_plei-1.png
</button>
</p>
<div id="fig-200exp_1000snp_cor_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/200exp_1000snp_cor_plei-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/200exp_1000snp_cor_plei-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now we see another odd thing – methods are not inflated even when
they have no first-stage factor analysis to infer confounders.</p>
<p>Finally, here’s both kinds of pleiotropy:</p>
<p><img src="figure/simulations.Rmd/200exp_1000snp_both_plei-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-200exp_1000snp_both_plei-1">
Past versions of 200exp_1000snp_both_plei-1.png
</button>
</p>
<div id="fig-200exp_1000snp_both_plei-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/200exp_1000snp_both_plei-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/200exp_1000snp_both_plei-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As usual, IVW and Robust drop in power, so methods like MR.Ash and
Susie perform relatively better here.</p>
<p>I ran some additional experiments (not shown here) which showed that
the lack of inflation under pleiotropy was due to a low number of
confounders relative to the number of exposures.</p>
<p>So the interesting result is that two ratios are important:</p>
<ul>
<li>The “M/K” or “num SNPs / num exposures” ratio – if &lt;1 than
traditional methods don’t work at all, and robust becomes inflated if
it’s close to 1.</li>
<li>The “J/K” or “num confounders / num exposures” ratio – if very low
(e.g. 1/100) then confounding will not inflate methods, even if Zs
affect many Xs. It is possible this can be overrided with very powerful
confounding effects.</li>
</ul>
</div>
<div id="simulations-using-uk-biobank-genotypes" class="section level2">
<h2>Simulations using UK Biobank Genotypes</h2>
<p>Here are some simulations using real genotypes from White European
samples in the UK Biobank. Phenotype simulations is still the same as
above – we just use real genotypes instead of simulating them. We
randomly downsample samples, SNPs, and loci from the mvSusie blood cell
trait data, which contains 975 loci averaging ~10k SNPs each and ~250k
samples. We usually use 100 loci, 1000 SNPs per locus, and 10-20k
samples.</p>
<p>In addition to using real genoytpes, we make these simulations more
realistic in another important dimension: SNP selection. Instead of
providing the true causal SNPs to the methods as in the above
simulations, we perform SNP selection with mvSusie, using the genotypes
and simulated phenotypes as input.</p>
<p>Here are the settings used for these simlations:</p>
<ul>
<li>10,000 samples</li>
<li>100 loci, each with 1-5 causal SNPs implanted (so ~300 overall)</li>
<li>30 exposures, 3 confounders</li>
<li><span class="math inline">\(\theta_{gx}=0.2\)</span>, <span
class="math inline">\(\theta_{gz}=0.4\)</span>, <span
class="math inline">\(\theta_{zx}=0.5\)</span>, <span
class="math inline">\(\theta_{zy}=0.375\)</span></li>
<li><span class="math inline">\(\phi_{gx}=0.05\)</span>, <span
class="math inline">\(\phi_{gy}=0.05\)</span>, <span
class="math inline">\(\phi_{gz}=0.05\)</span>, <span
class="math inline">\(\phi_{zx}=0.5\)</span></li>
<li>As usual, first four exposures are true effects with effect sizes
<span class="math inline">\(0.05,0.1,0.2,0.3\)</span></li>
</ul>
<p>Here are the results with correlated pleiotropy:</p>
<p><img src="figure/simulations.Rmd/cor_plei_ukbb_mvsusie-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_ukbb_mvsusie-1">
Past versions of cor_plei_ukbb_mvsusie-1.png
</button>
</p>
<div id="fig-cor_plei_ukbb_mvsusie-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_ukbb_mvsusie-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/cor_plei_ukbb_mvsusie-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And here are the results with both kinds of pleiotropy:</p>
<p><img src="figure/simulations.Rmd/both_plei_ukbb_mvsusie-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-both_plei_ukbb_mvsusie-1">
Past versions of both_plei_ukbb_mvsusie-1.png
</button>
</p>
<div id="fig-both_plei_ukbb_mvsusie-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/both_plei_ukbb_mvsusie-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/both_plei_ukbb_mvsusie-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="snp-selection-comparison" class="section level2">
<h2>SNP Selection Comparison</h2>
<p>How should we select SNPs to use as instruments? The traditional
strategy is to pick the SNP with the lowest p-value in each locus. But
this becomes tricky in the multi-exposure case. What if a locus has
different causal variants for different traits?</p>
<p>To test this, I compared several SNP selection strategies in the UK
Biobank simulations:</p>
<ul>
<li>mvSusie, i.e. the same strategy as above</li>
<li>mvSusie-RSS, the summary statistics mode of mvSusie</li>
<li>max: select the SNP at each locus that has the best z-score for any
trait</li>
<li>mean: select the SNP at each locus that has the best z-score,
averaged across all traits</li>
</ul>
<p>The latter three methods use summary statistics, which would normally
take an intractable amount of time to generate. So instead I simulated
the statistics from the RSS model,</p>
<p><span class="math display">\[\hat{\beta} \sim
\mathcal{N}(SRS\beta^{-1},SRS)\]</span></p>
<p>where the true <span class="math inline">\(\beta\)</span> are the
effects simulated by my phenotype simulator, the <span
class="math inline">\(R\)</span> is taken from the LD file included with
each mvSusie locus, and <span class="math inline">\(S\)</span> is a
matrix containing SNP standard deviations (see RSS paper). The standard
errors for the summary statistics were simply set to <span
class="math inline">\(1/\sqrt{N}\)</span>.</p>
<p>We can show that using mvSusie or mvSusie-RSS leads to a better F1
score for SNP selection, as well as improved downstream performance.</p>
<p>First, here is the precision / recall / F1 for each method (no
pleiotropy):</p>
<ul>
<li>mvSusie: 0.34358 / 0.47136 / 0.39688</li>
<li>mvSusie-RSS: 0.47689 / 0.48672 / 0.47989</li>
<li>max: 0.63289 / 0.2106 / 0.31591</li>
<li>mean: 0.42589 / 0.14216 / 0.21308</li>
</ul>
<p>Results look similar regardless of the level of pleiotropy. There is
a clear hierarchy, mvSusie-RSS &gt; mvSusie &gt; max &gt; mean, in terms
of F1. The high precision but low recall of max/mean reflects that they
can’t capture all causal SNPs at each locus. It’s a bit less clear why
mvSusie-RSS &gt; mvSusie, but it’s probably because I simulated the
sumstats exactly from the RSS model.</p>
<p>The next question, naturally, is whether this actually impacts
downstream results of the methods. The answer is yes, and here’s a graph
showing it for IVW and Susie (no pleiotropy):</p>
<p><img src="figure/simulations.Rmd/no_plei_selection-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no_plei_selection-1">
Past versions of no_plei_selection-1.png
</button>
</p>
<div id="fig-no_plei_selection-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/no_plei_selection-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/6c53a85fd9a18275b9374838c4408ff893282ffa/docs/figure/simulations.Rmd/no_plei_selection-1.png" target="_blank">6c53a85</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-08-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sims-with-true-genotypes-and-exposures-only-simulating-outcome"
class="section level2">
<h2>Sims with true genotypes and exposures, only simulating outcome</h2>
<p>In these simulations, we use true UK Biobank genotypes, as above, but
here we also use true exposure phenotype values. We only simulate the
outcome trait. Simulating the confounders is optional, but here I used
real confounders too. This makes the simulation as “realistic” as
possible, with the downside being that we don’t know which SNPs truly
affect the exposures &amp; can’t control those effects.</p>
<p>Here I used loci for 16 UK Biobank blood cell traits preprocessed for
the mvSusie paper. I downsampled to 50,000 samples and 200 loci for
simplicity. Out of the 16 traits, I picked two randomly to serve as
unobserved confounders.</p>
<p>Here are the results with no pleiotropy. Methods are calibrated, as
expected.</p>
<p><img src="figure/simulations.Rmd/no_plei_truegx-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no_plei_truegx-1">
Past versions of no_plei_truegx-1.png
</button>
</p>
<div id="fig-no_plei_truegx-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/no_plei_truegx-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/a1dd33895805194a34a0d3a45573b275c5e9daac/docs/figure/simulations.Rmd/no_plei_truegx-1.png" target="_blank">a1dd338</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now here are the results with correlated pleiotropy, with <span
class="math inline">\(\gamma_{zx} = \gamma_{zy} = 0.2\)</span>.
Remember, I don’t have control over <span
class="math inline">\(\gamma_{gz}\)</span>, since here <span
class="math inline">\(Z\)</span> are real traits. Here we see a great
deal of confounding, even with GFA:</p>
<p><img src="figure/simulations.Rmd/cor_plei_truegx-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor_plei_truegx-1">
Past versions of cor_plei_truegx-1.png
</button>
</p>
<div id="fig-cor_plei_truegx-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/cor_plei_truegx-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/a1dd33895805194a34a0d3a45573b275c5e9daac/docs/figure/simulations.Rmd/cor_plei_truegx-1.png" target="_blank">a1dd338</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>So we are getting results that seem plausible, but the confounding
seems quite strong. I ran more sims with <span
class="math inline">\(\gamma_{zx} = \gamma_{zy} = 0.1\)</span> (not
shown), which slightly reduced confounding, but methods were still
inflated even with GFA. Still need to figure out what’s going on
here.</p>
</div>
<div id="real-data-results" class="section level1">
<h1>Real data results</h1>
<p>I ran 2SLS/IVW/Robust/Susie/MR.Ash + nothing/GFA/RPCA on the loci for
16 UK Biobank blood cell traits preprocessed for the mvSusie paper. I
chose Hemoglobin as the outcome and the other 15 as exposures,
arbitrarily. I have no idea if this is reasonable, so take the
second-stage results with a grain of salt. I’ll update with more
realistic traits and better formatting later.</p>
<p>Here are the number of significant traits found by each method:</p>
<ul>
<li>2SLS: 10</li>
<li>2SLS-GFA: 10</li>
<li>2SLS-RPCA: 9</li>
<li>IVW: 10</li>
<li>IVW-GFA: 7</li>
<li>IVW-RPCA: 8</li>
<li>Robust: 7</li>
<li>Robust-GFA: 6</li>
<li>Robust-RPCA: 8</li>
<li>Susie: 3</li>
<li>Susie-GFA: 4</li>
<li>Susie-RPCA: 4</li>
<li>MR.Ash: 6</li>
<li>MR.Ash-GFA: 6</li>
<li>MR.Ash-RPCA: 6</li>
</ul>
<p>So we can see that sometimes first-stage methods reduce the number of
significant trait pairs. The methods vary on the number of exposures
found significant, though they generally agree on the order.</p>
<p>Here are the number of factors found by each FA method, and their
correlations with the 15 exposure traits (sorry for ugly raw
format):</p>
<ul>
<li>GFA: 4 factors (after pruning heuristic)</li>
</ul>
<pre><code>[1] &quot;Number of factors detected by GFA: 4&quot;
[1] &quot;Correlations with x_betas:&quot;
        WBC_count   RBC_count        MCV         RDW Platelet_count
[1,] -0.001640976 -0.75468968 0.98605224 -0.31663391     0.03624346
[2,]  0.325981371 -0.09764243 0.05087593 -0.05687534     0.93476251
[3,]  0.023064779 -0.07617768 0.12559695 -0.15648708     0.01058407
[4,] -0.321576727 -0.02835313 0.05563161 -0.00118629     0.03691328
     Plateletcrit         PDW Lymphocyte_perc Monocyte_perc Neutrophill_perc
[1,]   0.04016506 -0.06180007      0.06647548    0.05594493      -0.06597634
[2,]   0.90809002 -0.55897034      0.01045073    0.01053307      -0.03457303
[3,]   0.02770390  0.10704047     -0.06414842   -0.03178419       0.07043654
[4,]   0.00682206 -0.05983110      0.96675776    0.27570438      -0.97842300
     Eosinophill_perc Basophill_perc Reticulocyte_perc        MSCV     HLR_perc
[1,]      -0.06279681     0.02657169       0.102809344  0.88625102  0.170774450
[2,]       0.09842009     0.05558868      -0.008599868  0.03262786  0.007867302
[3,]      -0.01165736    -0.00792867       0.987386457 -0.14451179  0.987802241
[4,]       0.20059308     0.11994296      -0.082500786  0.12663053 -0.061190626</code></pre>
<ul>
<li>RPCA: 13 factors (no pruning)</li>
</ul>
<pre><code>[1] &quot;Number of factors detected by RPCA: 13&quot;
[1] &quot;Correlations with x_betas:&quot;
         WBC_count    RBC_count          MCV          RDW Platelet_count
 [1,] -0.102095006  0.462869586 -0.479733770  0.024229052   -0.372475422
 [2,]  0.419429298  0.284659704 -0.396475049  0.104518434    0.606220257
 [3,] -0.295600205  0.368873285 -0.420897576  0.171037432   -0.087991929
 [4,]  0.079700065 -0.049654160  0.070989330  0.089435198   -0.121957750
 [5,] -0.013486931  0.072250670  0.035881308  0.053908895   -0.059051433
 [6,] -0.408498335 -0.238159343  0.007617194 -0.380541389    0.103350258
 [7,] -0.232778402 -0.173081414 -0.056831128  0.308649364   -0.035678286
 [8,]  0.090406415 -0.126642244  0.042541912 -0.083316360   -0.001085696
 [9,] -0.059653731 -0.011806263 -0.052414418  0.270085345   -0.023874378
[10,]  0.014354183 -0.002924474  0.020023040  0.008973424    0.018205410
[11,]  0.010207131 -0.017109076  0.037672692 -0.010542113    0.013850411
[12,] -0.019022972  0.004579979  0.005743584  0.007400343    0.003209279
[13,]  0.004254445  0.008882500  0.010584905 -0.010955323    0.010688070
      Plateletcrit          PDW Lymphocyte_perc Monocyte_perc Neutrophill_perc
 [1,] -0.352704986  0.364713774    -0.361376169  -0.191987718      0.403521073
 [2,]  0.641283437 -0.285581415    -0.249106479  -0.060101095      0.224404739
 [3,] -0.147663273 -0.017198872     0.433218044   0.194557047     -0.493851833
 [4,] -0.135894160 -0.023659701    -0.444764748  -0.160943509      0.474331214
 [5,]  0.010917763  0.049580935    -0.269549980   0.442772389      0.001697879
 [6,] -0.032898149 -0.179395399    -0.056383699   0.029950428      0.042152127
 [7,]  0.093138535  0.206575311    -0.030301279   0.032833525      0.042485129
 [8,]  0.091555474  0.169117602    -0.036891945  -0.185170321     -0.007131183
 [9,] -0.074433455 -0.156936780    -0.018031752  -0.116356819      0.017082797
[10,]  0.015548812 -0.032775914    -0.003521535  -0.050230055      0.017867336
[11,]  0.004391547 -0.013130119    -0.008541238   0.037531366     -0.002805021
[12,] -0.007191359  0.009035615    -0.001589169  -0.006344848      0.004468541
[13,]  0.016166542  0.014291370    -0.005441077  -0.010111899      0.012487903
      Eosinophill_perc Basophill_perc Reticulocyte_perc         MSCV
 [1,]     -0.088915000   -0.117998027      0.2474364500 -0.601589489
 [2,]      0.059826493    0.012290744     -0.0962091162 -0.327354648
 [3,]      0.220223819    0.116363603     -0.4617865613 -0.226336510
 [4,]     -0.157427805   -0.069499723     -0.4847973241  0.187549039
 [5,]      0.335569245    0.168349777      0.0611062173  0.062306447
 [6,]     -0.016342239   -0.019982891      0.0658507416 -0.142766188
 [7,]     -0.154073560    0.193978437      0.0126234349  0.016055167
 [8,]      0.423262465    0.301601724     -0.0054947977 -0.005286929
 [9,]      0.210416415   -0.253647386      0.0003153864  0.003298467
[10,]      0.009229434    0.002925570      0.0661240105  0.001889833
[11,]     -0.001062457   -0.041301685      0.0623734135  0.001506438
[12,]     -0.005449317   -0.007815489      0.0123083211  0.002891452
[13,]     -0.013552033   -0.046556558      0.0173652680 -0.014306854
         HLR_perc
 [1,]  0.17455981
 [2,] -0.10057817
 [3,] -0.47560346
 [4,] -0.48563753
 [5,]  0.06599330
 [6,]  0.02873132
 [7,]  0.02948258
 [8,] -0.01456019
 [9,]  0.02461489
[10,]  0.06070806
[11,]  0.05783540
[12,]  0.01099033
[13,]  0.00724240</code></pre>
<div id="population-stratification-assessments" class="section level2">
<h2>Population stratification assessments</h2>
<p>Simulated population structure using Balding-Nichols model on two
populations. <span class="math inline">\(F_{ST}=0.1\)</span>, <span
class="math inline">\(\gamma_{psx}=0.25\)</span>, <span
class="math inline">\(\gamma_{psy}=0.25\)</span>, <span
class="math inline">\(\gamma_{psz}=0.25\)</span>.</p>
<p>No pleiotropy (just population stratification):</p>
<p><img src="figure/simulations.Rmd/no_plei_ps2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no_plei_ps2-1">
Past versions of no_plei_ps2-1.png
</button>
</p>
<div id="fig-no_plei_ps2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/no_plei_ps2-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/no_plei_ps2-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Both types of pleiotropy:</p>
<p><img src="figure/simulations.Rmd/both_plei_ps2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-both_plei_ps2-1">
Past versions of both_plei_ps2-1.png
</button>
</p>
<div id="fig-both_plei_ps2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/both_plei_ps2-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/both_plei_ps2-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here’s the power at a fixed FPR threshold, with no pleiotropy:</p>
<p><img src="figure/simulations.Rmd/powfpr_no_plei_ps2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-powfpr_no_plei_ps2-1">
Past versions of powfpr_no_plei_ps2-1.png
</button>
</p>
<div id="fig-powfpr_no_plei_ps2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/powfpr_no_plei_ps2-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>No pleiotropy (just population stratification), tougher setting:</p>
<p><img src="figure/simulations.Rmd/no_plei_ps4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no_plei_ps4-1">
Past versions of no_plei_ps4-1.png
</button>
</p>
<div id="fig-no_plei_ps4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/1d0517a5a4448399b6a462819524e85ef0d9bad1/docs/figure/simulations.Rmd/no_plei_ps4-1.png" target="_blank">1d0517a</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/nlapier2/mvmr_website/blob/08e3dae8978a7721593da93237f214f94ff76f27/docs/figure/simulations.Rmd/no_plei_ps4-1.png" target="_blank">08e3dae</a>
</td>
<td>
Nathan LaPierre
</td>
<td>
2023-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggpubr_0.6.0      ggplot2_3.5.0     DiagrammeR_1.0.10 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   xfun_0.39          bslib_0.5.0        purrr_1.0.1       
 [5] carData_3.0-5      colorspace_2.1-0   vctrs_0.6.5        generics_0.1.3    
 [9] htmltools_0.5.5    yaml_2.3.7         utf8_1.2.4         rlang_1.1.3       
[13] jquerylib_0.1.4    later_1.3.1        pillar_1.9.0       glue_1.7.0        
[17] withr_3.0.0        RColorBrewer_1.1-3 lifecycle_1.0.4    stringr_1.5.0     
[21] ggsignif_0.6.4     munsell_0.5.1      gtable_0.3.4       visNetwork_2.1.2  
[25] htmlwidgets_1.6.2  evaluate_0.21      labeling_0.4.3     knitr_1.43        
[29] callr_3.7.3        fastmap_1.1.1      httpuv_1.6.11      ps_1.7.5          
[33] fansi_1.0.6        highr_0.10         broom_1.0.5        Rcpp_1.0.12       
[37] backports_1.4.1    promises_1.2.0.1   scales_1.3.0       cachem_1.0.8      
[41] jsonlite_1.8.7     abind_1.4-5        farver_2.1.1       fs_1.6.3          
[45] digest_0.6.33      stringi_1.8.3      rstatix_0.7.2      processx_3.8.2    
[49] dplyr_1.1.2        getPass_0.2-2      rprojroot_2.0.3    grid_4.2.1        
[53] cli_3.6.2          tools_4.2.1        magrittr_2.0.3     sass_0.4.7        
[57] tibble_3.2.1       car_3.1-2          tidyr_1.3.0        whisker_0.4.1     
[61] pkgconfig_2.0.3    ellipsis_0.3.2     rmarkdown_2.23     httr_1.4.6        
[65] rstudioapi_0.15.0  R6_2.5.1           git2r_0.32.0       compiler_4.2.1    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
