---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r packages, echo=FALSE}
library("DiagrammeR")
```

Welcome to the MultiVariable Mendelian Randomization (MVMR) project website.
Here we briefly describe the project and provide some simulation results.
In the future we will also include instructions for running our software.

See [Metabolite MR](realdata.html) to see results from the analysis 
of real metabolite data, mostly from the UK Biobank and obtained via
the [IEU Open GWAS](https://gwas.mrcieu.ac.uk/datasets/) website.


### Project description

```{r dag, echo=FALSE}
DiagrammeR::grViz("digraph {
  graph [layout = dot, rankdir = LR]

  node [shape = circle]
  G [label = 'G']
  Z [label = 'Z']
  X [label = 'X']
  Y [label = 'Y']

  # edge definitions with the node IDs
  edge []
  G -> X [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GX</SUB></FONT>>]
  G -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GY</SUB></FONT>>]
  G -> Z [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GZ</SUB></FONT>>]
  X -> Z [dir=back; label=<\U03B8<FONT POINT-SIZE='8'><SUB>ZX</SUB></FONT>>]
  Z -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>ZY</SUB></FONT>>]
  X -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>XY</SUB></FONT>>]
  }",
  height = 350, width = 800)
```

In MVMR, we wish to determine which "exposure" phenotype(s) $X$ affect an "outcome" phenotype Y given genotypes $G$ that affect $X$. 
$X$ might be something like metabolites, gene expression, blood cell traits, etc.
$Y$ could be a disease or another cellular phenotype.
An example would be the classic question of whether HDL, LDL, and/or triglycerides ($X$) affect cardiovascular disease (CVD) risk ($Y$). 

However, if the genetic variants for $X$ affect $Y$ through pathways other than $X$, this can lead to false positive conclusions. 
Suppose $Z$ represents some gene whose expression level modulates both HDL level and CVD risk. 
Then even if HDL doesn't affect CVD risk, we may still find a false positive with standard MVMR methods.
This is called "confounding".
Specifically, when $Z$ is a phenotype caused by $G$ (as opposed to something like population stratification), this is called "(horizontal) pleiotropy".

This model is shown in the "Directed Acyclic Graph (DAG)" above.
The nodes represent the variables and the labels on the edges represent the effect size of one variable on another.
In the DAG above, the pathway from $G \rightarrow Z \rightarrow Y$ causes confounding, as does the direct path $G \rightarrow Y$, with the effect of this confounding controlled by the values $\theta_{GZ} * \theta_{ZY}$ and $\theta_{GY}$. 
The effect $\theta_{GZ} * \theta_{ZY}$ causes what is known as "correlated pleiotropy" since it is correlated with the exposures $X$ (since $\theta_{GZ}$ affects $X$), while the direct effect $\theta_{GY}$ is called "uncorrelated pleiotropy" since it is not correlated with $X$.

The causal model for the data given this DAG can be represented by a "Structural Equation Model (SEM)", which shows how each variable functionally affects another. The SEM for this DAG is

$$Z = G\theta_{GZ} + \epsilon_Z$$

$$X = G\theta_{GX} + Z\theta_{ZX} + \epsilon_X$$

$$Y = X\theta_{XY} + G\theta_{GY} + Z\theta_{ZY} + \epsilon_Y$$

Our challenge is to estimate $\theta_{XY}$ despite this confounding. Our method seeks to address this challenge by modeling potential confounders and adjusting for this unwanted signal in the data.


### Methods and Results

To evaluate existing approaches, I ran some [simulations](simulations.html) on data simulated from the SEM above. 
As expected, existing methods had either high False Positive Rates (FPRs) or very low power due to the confounding. 

How do we address this problem? The easiest thing would be if we already knew the true values of $\tilde{Z}$ (where $\tilde{Z} = G\theta_{GZ}$, ignoring the noise term) and $\theta_{GY}$. 
Then we could control for the confounding directly. 
Indeed, my [simulations](simulations.html) show that this "oracle" approach combined with the standard "two-stage least squares (2SLS)" estimation procedure controles FPR and maintains high power under arbitrarily strong confounding.

This is a good sanity check, but in the real world we will not know $\tilde{Z}$ or $\theta_{GY}$. 
How do we estimate them? Suppose for now that we have individual-level data available.
<!-- We leave aside estimating $\theta_{GY}$ for now, as this is generally an easier problem -- for example, "MR-Egger" is an existing solution to adjust for uncorrelated pleiotropy. -->
Let's start by focusing on $\tilde{Z}$.
In this case, since we already know $G$, the challenge is really to estimate $\theta_{GZ}$.
(An alternative that also works is to learn $\psi_X = \theta_{GZ} * \theta_{ZX}$).

One straightforward approach is to apply truncated singular value decomposition (SVD), or any other standard factor analysis (FA) method, to $\tilde{X}$. 
From our DAG / SEM, we have that $\tilde{X} = G * (\theta_{GX} + \theta_{GZ} * \theta_{ZX})$. 
We assume that each $\tilde{Z}$ affects several exposures. 
So our hope is that the latent factors will pick up on these factors affecting multiple variables.

However, these "one-stage" approaches (FA on $X$ without accounting for $Y$) are unlikely to work satisfactorily. 
One reason is that $\theta_{GZ}$, our target, is NOT independent of $Y$, since $Z$ affects $Y$ (by definition, since $Z$ is a confounder). 
Therefore, not accounting for $Y$ necessarily limits how well one can approximate $\theta_{GZ}$.

My [Gibbs sampler page](gibbs.html) shows the conditional distribution of $\theta_{GZ}$ given all other variables, assuming that they are all normally-distributed. 
This explicitly shows how $\theta_{GZ}$ depends on both X and Y. 
However, there are currently some limitations for the multiple-exposure case that I am working through, and I am not currently accounting for sparsity or non-normal unimodal distributions, like ash does.

For summary statistics, where $G$ is not available, we will probably have to perform FA on the betas and stderrs for $X$ and $Y$. 
One starting point is to assume that the direct effects of any given SNP on the exposures are independent, so that the betas for each exposure are only correlated due to the unobserved factors. 
We could explore extensions to existing Bayesian frameworks like RUV-Bayes, Flash, or Mash.
