---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r packages, echo=FALSE}
library("DiagrammeR")
```

Welcome to the MultiVariable Mendelian Randomization (MVMR) project website.
Here we briefly describe the project and provide some simulation results.
In the future we will also include instructions for running our software.

See [Metabolite MR](realdata.html) to see results from the analysis 
of real metabolite data, mostly from the UK Biobank and obtained via
the [IEU Open GWAS](https://gwas.mrcieu.ac.uk/datasets/) website.


## Project description

```{r dag, echo=FALSE}
DiagrammeR::grViz("digraph {
  graph [layout = dot, rankdir = LR]

  node [shape = circle]
  G [label = 'G']
  Z [label = 'Z']
  X [label = 'X']
  Y [label = 'Y']

  # edge definitions with the node IDs
  edge []
  G -> X [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GX</SUB></FONT>>]
  G -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GY</SUB></FONT>>]
  G -> Z [label=<\U03B8<FONT POINT-SIZE='8'><SUB>GZ</SUB></FONT>>]
  X -> Z [dir=back; label=<\U03B8<FONT POINT-SIZE='8'><SUB>ZX</SUB></FONT>>]
  Z -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>ZY</SUB></FONT>>]
  X -> Y [label=<\U03B8<FONT POINT-SIZE='8'><SUB>XY</SUB></FONT>>]
  }",
  height = 350, width = 800)
```

In MVMR, we wish to determine which "exposure" phenotype(s) $X$ affect an "outcome" phenotype Y given genotypes $G$ that affect $X$. 
$X$ might be something like metabolites, gene expression, blood cell traits, etc.
$Y$ could be a disease or another cellular phenotype.
An example would be the classic question of whether HDL, LDL, and/or triglycerides ($X$) affect cardiovascular disease (CVD) risk ($Y$). 

However, if the genetic variants for $X$ affect $Y$ through pathways other than $X$, this can lead to false positive conclusions. 
Suppose $Z$ represents some gene whose expression level modulates both HDL level and CVD risk. 
Then even if HDL doesn't affect CVD risk, we may still find a false positive with standard MVMR methods.
This is called "confounding".
Specifically, when $Z$ is a phenotype caused by $G$ (as opposed to something like population stratification), this is called "(horizontal) pleiotropy".

This model is shown in the "Directed Acyclic Graph (DAG)" above.
The nodes represent the variables and the labels on the edges represent the effect size of one variable on another.
In the DAG above, the pathway from $G \rightarrow Z \rightarrow Y$ causes confounding, as does the direct path $G \rightarrow Y$, with the effect of this confounding controlled by the values $\theta_{GZ} * \theta_{ZY}$ and $\theta_{GY}$. 
The effect $\theta_{GZ} * \theta_{ZY}$ causes what is known as "correlated pleiotropy" since it is correlated with the exposures $X$ (since $\theta_{GZ}$ affects $X$), while the direct effect $\theta_{GY}$ is called "uncorrelated pleiotropy" since it is not correlated with $X$.

The causal model for the data given this DAG can be represented by a "Structural Equation Model (SEM)", which shows how each variable functionally affects another. The SEM for this DAG is

$$Z = G\theta_{GZ} + \epsilon_Z$$

$$X = G\theta_{GX} + Z\theta_{ZX} + \epsilon_X$$

$$Y = X\theta_{XY} + G\theta_{GY} + Z\theta_{ZY} + \epsilon_Y$$

Our challenge is to estimate $\theta_{XY}$ despite this confounding. Our method seeks to address this challenge by modeling potential confounders and adjusting for this unwanted signal in the data.




## State of the project

Below, I summarize the narrative of the project, current results, and current
questions and things in progress.

#### Narrative

- High dimensional MR can help give granular causal insights into disease and complex trait etiology
- However, pleiotropic confounding is a big concern in MR. Few methods address it in the multivariate setting, and even fewer look at the high-dimensional case with possibly highly correlated traits.
- We demonstrate that factor analysis can be used to infer and correct for potential pleiotropic confounders, since

$\hat{\theta}_{GX} \approx \theta_{GX} + \theta_{GZ}\theta_{ZX},$

so we can run factor analysis on $\hat{\theta}_{GX}$ to estimate $\theta_{GZ}$,
i.e. the genetic component of confounders, and include them in the subsequent 
regression to control for them.

- We introduce FAMR, which is a general framework that breaks this problem into three main steps: (1) factor analysis; (2) polygenic modeling; (3) regression or variable selection.
- We propose using GFA for (1), susie or a PRS method for (2), and susie-rss for (3).
- The latter allows us to form credible sets of highly correlated exposures when present.
- cTWAS-style FAMR framework allows us to include SNPs and control for uncorrelated pleiotropy, and to handle sparsity in effects.
- We show that existing methods have inflated false positives under pleiotropy, or do not scale or perform well in the high dimensional setting (MVMR-cML).
- We also show that, as we include more and more exposures, because the SNP-exposure effect matrix gets sparser, power for traditional methods can decrease. Not so for FAMR.


#### Outline of method

1. Identify loci centered around genome-wide significant SNPs for each trait.

2. LD prune these SNPs to make memory/runtime feasible.

3. Run GFA on pruned SNPs. GFA's L_hat parameter is an estimate for $\theta_{GZ}$ and its F_hat is an estimate for $\theta_{ZX}$.

4. Learn SNP effect weights, i.e. sparsified effect estimates for SNPs on exposures and confounders, e.g. via multiple regression or summary statistics equivalent. Use susie, PRS, or other regression methods for this.

5. For both the exposures and inferred factors, impute Z-scores (for their effect on $Y$) and $R$ (correlations between these traits and between traits and SNPs). This follows the 
derivation in the [S-PrediXcan paper](https://www.nature.com/articles/s41467-018-03621-1), by way of cTWAS (see cTWAS supplementary information, equations 32-34).

$\hat{\zeta}_{X_jY} \approx \frac{w_j \hat{\zeta}_{GY}} {\sqrt{w_j R_j w_j^T}}$

where $j$ indexes the exposure, $w_j$ are weights (effect sizes) for the SNPs on exposure $j$, $R_j$ is the LD matrix of the SNPs, and $\hat{\zeta}_{GY}$ are the estimated Z-scores of the SNPs on the outcome trait.

Between a variant and a trait $j$,

$\hat{R}_{ij} = \frac{R_{i,j} w_j^T} {\sqrt{w_j R_j w_j^T}}$

where $R_{i,j}$ is the vector of correlations between SNP $i$ and the SNPs in
the prediction model of exposure $j$.

Between two traits $i$ and $j$,

$\hat{R}_{ij} = \frac{w_i R^s_{i,j} w_j^T} {\sqrt{w_i R_i w_i^T} \sqrt{w_j R_j w_j^T}}$

where $R_{i,j}$ is now defined as the correlation matrix between the SNPs in the
prediction models of exposures $i$ and $j$.


6. Run cTWAS-style framework to fit priors for SNPs and exposures+factors, output PIPs and credible sets. Essentially this involves repearted runs of $susie.rss(Z, R)$ (where $Z$ and $R$ come from above, and include both SNPs and traits), each time fitting prior inclusion probability and prior second moment parameters for each class of variable (SNP and exposure). See cTWAS paper for more details.



#### Current results

- Showing pleiotropy of existing methods and FA correcting for it: looks good in my standard simulations, both uncorrelated and correlated pleiotropy. (See FAMR Simulations page.)

- When doing simulations with stronger-effect confounders, didn't do quite as well (need to upload these results)

- Showing loss of power in high dimensions due to inclusion of many irrelevant/weak instruments -- had some initial results on this that looked promising (see overleaf), but need to update. In-progress.

- Real data: FAMR's results seem very underpowered when regressing out factors. When not regressing them out the power is better, but still not great, and has a few questionable results. (See Metabolite MR page.)

- Real data with highly correlated exposures (lipoprotein subfractions): early results looked reasonable; need to clean and upload.



#### Questions / In-progress

- As mentioned above, working on showing robustness to power loss or false positives with many exposures.

- A new strategy I am pursuing is, instead of heavily LD pruning before FAMR, instead only heavily prune SNPs for GFA (since GFA does not model LD). Then use GFA to impute L_hat (i.e. $\theta_{GZ}$) for all SNPs, using the learned F_hat (i.e. $\theta_{ZX}$, which does not depend on SNPs). Then the Z and R are built on all SNPs (or a much larger subset) and thus should hopefully be more accurate. I have implemented this and I am currently assessing it.

- How can we improve the estimation of R between exposures and factors? Would better (possibly PRS-based) weight-learning schemes help? Would this fix issues with false positive control when not annihilating factors?

- Running simulations that use the true G-X signal from real data, as well as factors inferred by GFA, only simulating Y. So far, some odd results, but need to debug.

- Regressing out factors vs keeping them in -- maybe keep the former as a conservative approach, but latter has more power, even if it doesn't control FPR quite as well.

- There are still memory usage improvements I am working on.

- Maybe more time consuming / future work: can we incorporate uncertainty about weight estimates (and thus Z and R) into susie-rss, to avoid the problem of confidently selecting with 100% PIP one of two highly-correlated exposures, and instead put them in a credible set together?


