---
title: "Metabolite MR"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r packages, echo=FALSE, include=FALSE}
library(data.table)
library(dplyr)
library(knitr)
library(ggplot2)
library(reshape2)
```

```{r process_dat, echo=FALSE}
# read in data and select methods to display
dat = fread('data/summary_of_results_annih_max.tsv', sep='\t', header=T)
dat = dat %>% select(Exposure, Outcome, famr_none, famr_gfa, ivw, robust, ivw.univar)
# sort by average PIP across all methods
dat$Average = rowSums(dat %>% select(famr_none, famr_gfa, ivw, robust, ivw.univar)) / (ncol(dat) - 2)
dat = dat[order(dat$Average, decreasing=TRUE), ]
dat$Average = NULL 
 
# read in data and select methods to display
dat2 = fread('data/summary_of_results_noannih_clumped.tsv', sep='\t', header=T)
dat2 = dat2 %>% select(Exposure, Outcome, famr_none, famr_gfa, ivw, robust, ivw.univar)
# sort by average PIP across all methods
dat2$Average = rowSums(dat2 %>% select(famr_none, famr_gfa, ivw, robust, ivw.univar)) / (ncol(dat2) - 2)
dat2 = dat2[order(dat2$Average, decreasing=TRUE), ]
dat2$Average = NULL 

# read in data and select methods to display
dat_rsp = fread('data/summary_of_rsparsepro_wts_susie_final_results.tsv', sep='\t', header=T)
dat_rsp = dat_rsp %>% select(Exposure, Outcome, famr_none, famr_gfa, ivw, robust, ivw.univar)
# sort by average PIP across all methods
dat_rsp$Average = rowSums(dat_rsp %>% select(famr_none, famr_gfa, ivw, robust, ivw.univar)) / (ncol(dat_rsp) - 2)
dat_rsp = dat_rsp[order(dat_rsp$Average, decreasing=TRUE), ]
dat_rsp$Average = NULL 

# read in data and select methods to display
dat_rescalez = fread('data/summary_of_rescale_z_results.tsv', sep='\t', header=T)
dat_rescalez = dat_rescalez %>% select(Exposure, Outcome, famr_none, famr_gfa, ivw, robust, ivw.univar)
# sort by average PIP across all methods
dat_rescalez$Average = rowSums(dat_rescalez %>% select(famr_none, famr_gfa, ivw, robust, ivw.univar)) / (ncol(dat_rescalez) - 2)
dat_rescalez = dat_rescalez[order(dat_rescalez$Average, decreasing=TRUE), ]
dat_rescalez$Average = NULL 
```

In order to assess the performance of FAMR in a real data setting,
we performed a high-dimensional / multi-variable Mendelian Randomization (MR)
analysis of the effect of various metabolic and anthropometric traits on several 
complex diseases (coronary artery disease, stroke, type 2 diabetes, and asthma).


## Data Sources

Summary statistics for the exposure traits were obtained mostly from the 
UK Biobank while the outcome traits were obtained from other sources, so the
analysis is two-sample MR.
All summary statistics were obtained from the
[IEU Open GWAS](https://gwas.mrcieu.ac.uk/datasets/)
website.

The 33 exposure traits chosen are the union of the 20 exposure traits analyzed by 
[Karjalainen et al](https://doi.org/10.1038/s41586-024-07148-y)
in their MR analysis and those analyzed by 
[Morrison et al](https://doi.org/10.1038/s41588-020-0631-4)
in their analysis. 
We excluded Glycerol because there is no large (>100k samples) dataset available
on the IEU Open GWAS portal.
We also included Apolipoprotein B (ApoB) and Poly-Unsaturated Fatty Acids (PUFA)
as these are known to play a causal role in coronary artery disease (CAD).

[Morrison et al](https://www.nature.com/articles/s41588-020-0631-4/tables/1)
provide a helpful table of which of the exposures they analyzed are expected
to affect each of the outcomes based on literature support.

Here are the exposures used in the analysis:

```{r exposures}
sort(unique(dat$Exposure))
```


## Processing Pipeline

Preprocessing the data consisted of the following steps:

1. For each exposure, we centered 1 Mb windows around each SNP with a p-value
of less than $5*10^{-8}$ (standard genome-wide significance).

2. We then merged these regions across all exposures, e.g. overlapping regions
were merged together.

3. We obtained the LD matrix of the SNPs in each of these regions using 
individual-level data from the UK Biobank.

4. SNPs that were not in the UK Biobank, had minor allele frequency below 0.01,
were multi-allelic, had an NA value, or were in the Human Leukocyte Antigen (HLA)
region were filtered out.

5. Due to the high memory requirements, we performed LD clumping to reduce the 
total number of SNPs, with a threshold of $|r|=0.1$. In particular, SNPs were
sorted by increasing p-value, and SNPs with abs(LD) greater than 0.1 with a SNP
with a stronger p-value were removed. This also improves the performance of 
most factor analysis methods and MVMR methods in simulations.

6. We ran FAMR and various other MVMR methods (see table below) on this data.
We also ran univariate IVW MR ("ivw.univar") on each exposure-outcome pair.
For FAMR, we ran it both with and without regressing out ("annihilating")
the learned factors from the exposures.


## Trait names vs. codes

Consult the following list for trait names vs their codes.
I have tried to put the trait name in the results below where possible,
but this will be useful in cases where I forgot or it was impractical.

<details><summary>Click to expand</summary>

```
Trait Name	                     Trait ID
Alcohol	                         ieu-b-73
Birth Weight (BW)	               ieu-a-1083
Body Fat	                       ieu-a-999
Body Mass Index	                 ieu-a-2
Diastolic Blood Pressure	       ieu-b-39
Fasting Glucose	                 ieu-b-113
Height	                         ieu-a-89
High Density Lipoprotein (HDL)	 ieu-a-299
Low Density Lipoprotein (LDL)	   ieu-a-300
Smoking	                         ieu-b-4877
Systolic Blood Pressure	         ieu-b-38
Triglycerides	                   ieu-a-302
Asthma	                         ebi-a-GCST006862
Coronary Artery Disease (CAD)	   ebi-a-GCST005195
Stroke	                         ebi-a-GCST005838
Type 2 Diabetes	                 ieu-a-26
Acetate	                         met-d-Acetate
Acetone	                         met-d-Acetone
Alanine	                         met-d-Ala
Albumin	                         met-d-Albumin
3-hydroxybutyrate	               met-d-bOHbutyrate
Citrate	                         met-d-Citrate
Creatinine	                     met-d-Creatinine
Glucose	                         met-d-Glucose
Glutamine	                       met-d-Gln
Glycoprotein acetylation	       met-d-GlycA
Glycine	                         met-d-Gly
Histidine	                       met-d-His
Isoleucine	                     met-d-Ile
Lactate	                         met-d-Lactate
Leucine	                         met-d-Leu
Phenylalanine	                   met-d-Phe
Pyruvate	                       met-d-Pyruvate
Tyrosine	                       met-d-Tyr
Valine	                         met-d-Val
Apolipoprotein B	               met-d-ApoB
Polyunsaturated Fatty Acids	     met-d-PUFA
```

</details>


## Characterization of the data

Here we discuss the number of independent loci and heritability of each trait,
as well as its most genetically-correlated traits.

#### Number of regions

Across all 33 exposure traits, 2530 1 Mb loci were identified (avg. about 77 per trait),
and there were 726 independent loci after merging overlapping loci.
Excluding blood pressure and height, which were by far the most polygenic,
there were 1097 regions across 30 traits (avg. about 37 per trait).=,
and there were 477 independent loci after merging overlapping loci.

Here are the number of independent 1 Mb loci per trait:

<details><summary>Click to expand</summary>

```
-   522 regions_combined/ieu-b-39_regions.txt (Diastolic blood pressure)
-   495 regions_combined/ieu-b-38_regions.txt (Systolic blood pressure)
-   416 regions_combined/ieu-a-89_regions.txt (Height)
-   103 regions_combined/ieu-b-4877_regions.txt (Smoking status)
-    86 regions_combined/ieu-a-299_regions.txt (HDL cholesterol)
-    83 regions_combined/ieu-a-2_regions.txt (Body mass index)
-    69 regions_combined/met-d-Creatinine_regions.txt (Creatinine)
-    67 regions_combined/ieu-a-300_regions.txt (LDL cholesterol)
-    55 regions_combined/met-d-PUFA_regions.txt (Polyunsaturated fatty acids)
-    55 regions_combined/ieu-a-302_regions.txt (Triglycerides)
-    53 regions_combined/ieu-a-1083_regions.txt (Birth weight)
-    48 regions_combined/met-d-GlycA_regions.txt (Glycoprotein acetylation)
-    47 regions_combined/met-d-Gln_regions.txt (Glutamine)
-    46 regions_combined/met-d-ApoB_regions.txt (Apolipoprotein B)
-    41 regions_combined/met-d-Gly_regions.txt (Glycine)
-    39 regions_combined/ieu-b-73_regions.txt (Alcohol intake)
-    32 regions_combined/met-d-Citrate_regions.txt (Citrate)
-    32 regions_combined/met-d-Albumin_regions.txt (Albumin)
-    32 regions_combined/met-d-Ala_regions.txt (Alanine)
-    28 regions_combined/met-d-Tyr_regions.txt (Tyrosine)
-    23 regions_combined/met-d-Glucose_regions.txt (Glucose)
-    21 regions_combined/ieu-b-113_regions.txt (Fasting glucose)
-    19 regions_combined/met-d-Pyruvate_regions.txt (Pyruvate)
-    18 regions_combined/met-d-Val_regions.txt (Valine)
-    15 regions_combined/met-d-Leu_regions.txt (Leucine)
-    15 regions_combined/met-d-His_regions.txt (Histidine)
-    14 regions_combined/met-d-bOHbutyrate_regions.txt (bOHbutyrate)
-    11 regions_combined/met-d-Acetone_regions.txt (Acetone)
-    10 regions_combined/met-d-Lactate_regions.txt (Lactate)
-    10 regions_combined/ieu-a-999_regions.txt (Body fat)
-     9 regions_combined/met-d-Acetate_regions.txt (Acetate)
-     8 regions_combined/met-d-Phe_regions.txt (Phenylalanine)
-     8 regions_combined/met-d-Ile_regions.txt (Isoleucine)
```

</details>


#### Heritability

We used the [HDL package](https://github.com/zhenin/HDL) to compute heritability
and genetic correlations.

Below is the heritability of each trait (including the outcomes).
As you can see, except for the highly polygenic blood pressure and height traits,
all others have heritabilities between $1\%$ and $27\%$, with most being in the
$5\%$ to $18\%$ range. The median heritability is about $9\%$.

<details><summary>Click to expand</summary>

```
               trait                   h2
           ieu-b-38 (SBP)              0.7654
           ieu-b-39 (DBP)              0.7616
           ieu-a-89 (Height)           0.5236
         ieu-b-4877 (Smoking)          0.2612
          met-d-Gly (Glycine)          0.2506
            ieu-a-2 (BMI)              0.1788
        met-d-GlycA (Glyc. Acet.)      0.1742
           ieu-b-73 (Alc)              0.1709
   ebi-a-GCST005195 (CAD)              0.1667
  met-d-Creatinine                     0.1560
        met-d-PUFA                     0.1524
         met-d-Gln (Glutamine)         0.1345
        ieu-a-1083 (Birth Wt.)         0.1288
         met-d-Tyr (Tyrosine)          0.1133
        met-d-ApoB                     0.1097
     met-d-Citrate                     0.1031
     met-d-Albumin                     0.0965
         met-d-Ala (Alanine)           0.0900
         met-d-Val (Valine)            0.0885
         ieu-a-302 (Trigly.)           0.0877
         ieu-a-299 (HDL)               0.0840
         met-d-Leu (Leucine)           0.0704
     met-d-Glucose                     0.0695
    met-d-Pyruvate                     0.0649
 met-d-bOHbutyrate                     0.0559
         met-d-Ile (Isoleuc.)          0.0547
         ieu-a-300 (LDL)               0.0537
     met-d-Acetone                     0.0524
     met-d-Lactate                     0.0516
     met-d-Acetate                     0.0479
         met-d-Phe (Phenylal.)         0.0459
         met-d-His (Histidine)         0.0385
  ebi-a-GCST006862 (Asthma)            0.0342
         ieu-a-999 (Body fat)          0.0281
          ieu-a-26 (T2D)               0.0243
  ebi-a-GCST005838 (Stroke)            0.0235
         ieu-b-113 (Fast. Gluc.)       0.0145
```

</details>


#### Genetic correlation

Below are the genetic correlations between traits computed by the 
[HDL package](https://github.com/zhenin/HDL).

Most of them look reasonable, but some are obviously over-estimates
(e.g. some correlations are over 1).
One big surprise is that HDL and LDL are hardly correlated.

About half of the traits have below 50% correlation with all other traits
- Most of the non-BCAA amino acids
- Ketone bodies
- Glycolysis-related metabolites
- Fluid balance traits

The other roughly half of traits form small highly-correlated clusters. Examples:
- BCAAs (Valine, Leucine, Isoleucine)
- LDL, ApoB, and PUFA
- Triglycerides and GlycA
- T2D, Fasting Glucose, Glucose
- BMI and Body Fat
- Systolic and Diastolic blood pressure
- CAD and Stroke


Here are the top five correlations for each trait:

<details><summary>Click to expand</summary>

```
Genetic correlations for ebi-a-GCST005195: 

               trait    corr
1   ebi-a-GCST005838  0.7797
2           ieu-a-26  0.4230
3           ieu-b-38  0.3812
4          ieu-a-299 -0.3723
5          ieu-a-999  0.3710



Genetic correlations for ebi-a-GCST005838: 

               trait    corr
1   ebi-a-GCST005195  0.7797
2           ieu-a-26  0.7256
3           ieu-b-38  0.5899
4          ieu-b-113  0.5394
5          ieu-a-999  0.5265



Genetic correlations for ebi-a-GCST006862: 

               trait    corr
1           ieu-a-26  0.2385
2          ieu-a-999  0.1635
3      met-d-Lactate  0.1595
4      met-d-Acetate -0.1492
5      met-d-Acetone -0.1365



Genetic correlations for ieu-a-1083: 

               trait       corr
1           ieu-a-89  0.4254000
2           ieu-a-26 -0.2863000
3           ieu-b-38 -0.2067000
4   ebi-a-GCST005838 -0.1901000
5          ieu-b-113 -0.1881000




Genetic correlations for ieu-a-26: 

               trait    corr
1          ieu-b-113  1.1869
2   ebi-a-GCST005838  0.7256
3          met-d-Val  0.6845
4          met-d-Ile  0.6437
5          met-d-Leu  0.5972




Genetic correlations for ieu-a-299: 

               trait    corr
1          ieu-a-302 -0.6848
2          met-d-Leu -0.5278
3          met-d-Ile -0.5216
4          met-d-Val -0.4811
5        met-d-GlycA -0.4504




Genetic correlations for ieu-a-2: 

               trait    corr
1          ieu-a-999  1.0984
2           ieu-a-26  0.4698
3          met-d-Ile  0.3667
4          ieu-b-113  0.3641
5          met-d-Val  0.3465




Genetic correlations for ieu-a-300: 

               trait    corr
1         met-d-ApoB  1.1036
2         met-d-PUFA  0.8297
3          ieu-a-302  0.4555
4        met-d-GlycA  0.3966
5   ebi-a-GCST005195  0.2779




Genetic correlations for ieu-a-302: 

               trait    corr
1        met-d-GlycA  0.7566
2          ieu-a-299 -0.6848
3          met-d-Val  0.6040
4          met-d-Ile  0.6016
5          met-d-Leu  0.5518




Genetic correlations for ieu-a-89: 

               trait    corr
1         ieu-a-1083  0.4254
2      met-d-Lactate -0.1501
3      met-d-Acetate  0.1385
4        met-d-GlycA -0.1158
5          ieu-a-999 -0.1154




Genetic correlations for ieu-a-999: 

               trait    corr
1            ieu-a-2  1.0984
2           ieu-a-26  0.5278
3   ebi-a-GCST005838  0.5265
4        met-d-GlycA  0.5159
5          met-d-Phe  0.5021




Genetic correlations for ieu-b-113: 

               trait    corr
1      met-d-Glucose  1.3238
2           ieu-a-26  1.1870
3   ebi-a-GCST005838  0.5394
4          met-d-Ala  0.3701
5            ieu-a-2  0.3641




Genetic correlations for ieu-b-38: 

               trait    corr
1           ieu-b-39  0.8183
2   ebi-a-GCST005838  0.5899
3   ebi-a-GCST005195  0.3812
4           ieu-a-26  0.2913
5          ieu-b-113  0.2854




Genetic correlations for ieu-b-39: 

               trait    corr
1           ieu-b-38  0.8183
2   ebi-a-GCST005838  0.5168
3   ebi-a-GCST005195  0.3087
4           ieu-a-26  0.2087
5         ieu-a-1083 -0.1685




Genetic correlations for ieu-b-4877: 

               trait    corr
1           ieu-b-73  0.4361
2   ebi-a-GCST005838  0.3304
3   ebi-a-GCST005195  0.2555
4          ieu-a-999  0.2368
5        met-d-GlycA  0.2131




Genetic correlations for ieu-b-73: 

               trait       corr
1         ieu-b-4877  0.4361000
2          ieu-a-299  0.2524000
3      met-d-Acetone  0.2249000
4      met-d-Citrate -0.2223000
5      met-d-Acetate  0.1415000




Genetic correlations for met-d-Acetate: 

               trait    corr
1      met-d-Lactate -0.5745
2        met-d-GlycA -0.5055
3   ebi-a-GCST005838 -0.4637
4          ieu-a-999 -0.4556
5          ieu-a-302 -0.4030




Genetic correlations for met-d-Acetone: 

               trait    corr
1  met-d-bOHbutyrate  0.5463
2        met-d-GlycA -0.3522
3          ieu-a-302 -0.3022
4          met-d-Ala -0.2856
5      met-d-Acetate  0.2843




Genetic correlations for met-d-Ala: 

               trait    corr
1          met-d-Ile  0.4868
2          met-d-Leu  0.4491
3          met-d-Val  0.4450
4           ieu-a-26  0.4261
5          ieu-b-113  0.3701




Genetic correlations for met-d-Albumin: 

               trait    corr
1          ieu-a-999 -0.4238
2          met-d-Phe -0.3036
3            ieu-a-2 -0.2880
4   ebi-a-GCST005838 -0.2476
5          met-d-His  0.1864




Genetic correlations for met-d-ApoB: 

               trait    corr
1          ieu-a-300  1.1036
2         met-d-PUFA  0.7652
3          ieu-a-302  0.4816
4        met-d-GlycA  0.2698
5   ebi-a-GCST005838 -0.2280




Genetic correlations for met-d-bOHbutyrate: 

              trait    corr
1     met-d-Acetone  0.5463
2         ieu-a-999  0.2629
3     met-d-Lactate  0.2563
4         met-d-Ala -0.2461
5         met-d-Tyr -0.2204




Genetic correlations for met-d-Citrate: 

               trait    corr
1   met-d-Creatinine  0.2538
2           ieu-b-73 -0.2223
3         ieu-b-4877 -0.1716
4      met-d-Acetone  0.1354
5  met-d-bOHbutyrate  0.1317




Genetic correlations for met-d-Creatinine: 

               trait    corr
1          ieu-a-302  0.3592
2        met-d-GlycA  0.2742
3      met-d-Citrate  0.2539
4          ieu-a-299 -0.2244
5          met-d-Ile  0.2223




Genetic correlations for met-d-Gln: 

               trait    corr
1          met-d-His  0.4011
2          ieu-a-999 -0.2858
3      met-d-Acetate  0.2532
4          ieu-a-302 -0.2433
5      met-d-Lactate -0.2381




Genetic correlations for met-d-Glucose: 

               trait    corr
1          ieu-b-113  1.3238
2           ieu-a-26  0.5720
3      met-d-Lactate -0.2855
4        met-d-GlycA  0.2739
5          ieu-a-999  0.2482




Genetic correlations for met-d-GlycA: 

               trait    corr
1          ieu-a-302  0.7566
2          ieu-a-999  0.5159
3      met-d-Acetate -0.5055
4           ieu-a-26  0.4640
5          ieu-a-299 -0.4504




Genetic correlations for met-d-Gly: 

               trait    corr
1           ieu-a-26 -0.3062
2          ieu-a-999 -0.2387
3          ieu-b-113 -0.2232
4          met-d-Val -0.2222
5          met-d-Gln  0.1838




Genetic correlations for met-d-His: 

               trait    corr
1          met-d-Gln  0.4011
2   ebi-a-GCST005838 -0.2631
3          met-d-Ala  0.2453
4      met-d-Lactate -0.2362
5          ieu-a-999 -0.2301




Genetic correlations for met-d-Ile: 

               trait    corr
1          met-d-Leu  0.9522
2          met-d-Val  0.9351
3           ieu-a-26  0.6437
4          ieu-a-302  0.6016
5          ieu-a-299 -0.5216




Genetic correlations for met-d-Lactate: 

               trait    corr
1      met-d-Acetate -0.5745
2     met-d-Pyruvate  0.3740
3           ieu-a-26  0.3611
4          ieu-a-302  0.3313
5      met-d-Glucose -0.2855




Genetic correlations for met-d-Leu: 

               trait    corr
1          met-d-Ile  0.9522
2          met-d-Val  0.9275
3           ieu-a-26  0.5972
4          ieu-a-302  0.5518
5          ieu-a-299 -0.5278




Genetic correlations for met-d-Phe: 

               trait    corr
1          met-d-Ile  0.5117
2          ieu-a-999  0.5021
3          met-d-Tyr  0.4812
4           ieu-a-26  0.4648
5          met-d-Leu  0.4629




Genetic correlations for met-d-PUFA: 

               trait    corr
1          ieu-a-300  0.8297
2         met-d-ApoB  0.7652
3          ieu-a-302  0.5174
4        met-d-GlycA  0.3093
5          ieu-a-299  0.2889




Genetic correlations for met-d-Pyruvate: 

               trait    corr
1      met-d-Lactate  0.3740
2   ebi-a-GCST005838  0.2643
3      met-d-Acetate -0.2584
4        met-d-GlycA  0.2499
5          ieu-a-999  0.2409




Genetic correlations for met-d-Tyr: 

               trait    corr
1          met-d-Phe  0.4812
2          met-d-Ile  0.3482
3          met-d-Val  0.3107
4          met-d-Ala  0.3042
5           ieu-a-26  0.3029




Genetic correlations for met-d-Val: 

               trait    corr
1          met-d-Ile  0.9351
2          met-d-Leu  0.9275
3           ieu-a-26  0.6845
4          ieu-a-302  0.6041
5          ieu-a-299 -0.4811
```

</details>




## Results

The results table is below, sorted by decreasing average probability
(across all methods)
of the exposure having a causal effect on the outcome. 
For FAMR, the probability of being causal is given
by the posterior inclusion probability (PIP). 
For other methods, we performed false discovery rate (FDR) control across
all exposures for each outcome trait using the Q-value method.
The probability of causality was computed as 1 minus the Q-value.

In general, in line with simulation results, we
observed that FAMR is a bit more conservative than IVW. It occasionally misses
a trait pair that IVW gets that has good evidence according to
[Morrison et al](https://www.nature.com/articles/s41588-020-0631-4/tables/1).
However, it also avoids some likely spurious positives that IVW picks up.
It should be kept in mind that some of the positives identified by FAMR are
factors, and by examining the factor loadings we can see that some of the
exposure-outcome pairs that are likely causal are loaded on these factors.

In turn, univariate IVW has more positives than MVMR-IVW. 
Part of this is because it does not control for other exposures.
As a classic illustration, univariate IVW finds that HDL cholesterol causes
CAD -- a finding that is considered a false positive -- while no MVMR methods do,
including FAMR.

<details><summary>Click to expand</summary>

```{r print_results}
knitr::kable(dat)
```

</details>



## Results with no factor annihilation and clumped SNP selection

Here we present a slightly different run setting, with SNPs for non-FAMR
MVMR methods being selected by simply LD clumping SNPs
(again at a threshold of 0.1, so using the same set of SNPs as FAMR)
rather than only taking the max Z-score SNP per locus.
Also, we ran FAMR with no factor annihilation.

As we can see, FAMR with no factor annihilation is less conservative than the
one with factor annihilation.
MVMR methods in the clumped setting had many more positive findings,
some of which are likely true but many of which are likely false, including
finding an effect of HDL on CAD independent of LDL.

<details><summary>Click to expand</summary>

```{r print_results2}
knitr::kable(dat2)
```

</details>



## Results with rescaled Z-scores

Sometimes there have been issues with very high
Z-scores for exposures computed based on susie weights when there were many SNPs.
Also, the performance of the method in both simulations and real data tends to be
unstable when there are very high Z-scores. Thus, I tried running some runs
with exposure z-scores scaled down (equally) so that the maximum
z-score among all exposures was 20.


The results for this are shown below, with only CAD and T2D for brevity.
This does seem to avoid a lot of false positives for CAD, but there are some
iffy results for T2D, including Citrate and Albumin.

<details><summary>Click to expand</summary>

```{r print_rescale_z_results}
knitr::kable(dat_rescalez)
```

</details>



## Results with RSparsePro

[RSparsePro](https://doi.org/10.1101/2024.10.29.620968) is a method that is like
Susie-RSS, but is supposed to be more robust to errors in the R matrix between
exposures, factors, and SNPs.
Since that has been a challenge for our method, I decided to try running
RSparsePro in place of Susie-RSS.
Specifically, RSparsePro is used to compute the SNP-exposure weights used to
compute the Z-scores and R between exposures and factors, while I still use the
susie/cTWAS framework for the final regression to generate the PIPs for traits.

This produced a sensible, if slightly conservative, set of results across
the four outcomes tested.

<details><summary>Click to expand</summary>

```{r print_rsparsepro_results}
knitr::kable(dat_rsp)
```

</details>



## CAD Z-scores, PIPs, and credible set indices between FAMR versions

This table compiles the Z-scores, PIPs, and credible set indices learned by
four different versions of FAMR on CAD.

<details><summary>Click to expand</summary>

```{r print_cad_z_tab}
z_tab = data.frame(fread('data/mvmr-cad-realdata-combined-zscores-pips.tsv', sep='\t', header=T), row.names=1)
colnames(z_tab) = c("No FA, learned weights", "No annihilation, learned weights", 
                    "No annihilation, weights = betas", "Annihilator mode, learned weights",
                    "No annihilation, RSparsePro weights")
knitr::kable(z_tab)
```

</details>



## Correlations between exposures and factors

Here we give the exposure-exposure and exposure-factor correlations based on 
either weights learned by susie or weights = betas (effect size estimates from GWAS).

We will see that, in general, exposure-factor and exposure-exposure correlations
are estimated to be lower with weights learned by susie than when using betas as weights.

Here are my interpretations of the factors, based on their estimated correlations with exposures:

* Factor 1 is clearly strongly associated with systolic and diastolic blood pressure,
and thus can be thought of as a general "blood pressure factor".
* Factor 2 is strongly associated with leucine, isoleucine, and valine, and thus
can be thought of as a "branched-chain amino acids" factor.
* For the most part, factors 3-8 have more modest associations with exposures.
* Factor 3 is associated primarily with PUFA and ApoB, which are CAD risk factors.
* Factor 7 is primarily positively associated with triglycerides, LDL, and BMI,
and negatively with HDL.
* Factor 8 is primarily associated with the lifestyle traits of smoking and alcohol.
* Factors 5 and 6 are modestly associated with a number of ketone-containing traits,
e.g. Acetone, bOHbutyrate, Pyruvate.
* Factor 4 is difficult for me to interpret, as it is modestly associated with
a wide variety of traits.


```{r read_R, echo=FALSE}
R_learnwts = readRDS('data/results_famr_gfa_noannih_ebi-a-GCST005195.rds')
R_learnwts = as.matrix(R_learnwts$R)[1:41,1:41]
colnames(R_learnwts) = rownames(z_tab)
rownames(R_learnwts) = rownames(z_tab)

R_nolearnwts = readRDS('data/results_famr_gfa_nolearnwts_ebi-a-GCST005195.rds')
R_nolearnwts = as.matrix(R_nolearnwts$R)[1:41,1:41]
colnames(R_nolearnwts) = rownames(z_tab)
rownames(R_nolearnwts) = rownames(z_tab)

R_rsparsepro = readRDS('data/results_famr_gfa_rsparsepro_ebi-a-GCST005195.rds')
R_rsparsepro = as.matrix(R_rsparsepro$R)[1:41,1:41]
colnames(R_rsparsepro) = rownames(z_tab)
rownames(R_rsparsepro) = rownames(z_tab)
```

Here are the learned exposure-factor correlations with weights learned by susie:

```{r exp_fac_cor_learnwts, echo=FALSE}
data_learnwts <- melt(R_learnwts[1:33, 34:41])
ggplot(data_learnwts, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Here are the learned exposure-factor correlations learned by RSparsePro:

```{r exp_fac_cor_rsparsepro, echo=FALSE}
data_rsparsepro <- melt(R_rsparsepro[1:33, 34:41])
ggplot(data_rsparsepro, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Here are the learned exposure-factor correlations with weights = betas:

```{r exp_fac_cor_nolearnwts, echo=FALSE}
data_nolearnwts <- melt(R_nolearnwts[1:33, 34:41])
ggplot(data_nolearnwts, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

-------

Now here are the exposure-exposure correlations with weights learned by susie:

```{r exp_exp_cor_learnwts, echo=FALSE}
data_learnwts2 <- melt(R_learnwts[1:33, 1:33])
ggplot(data_learnwts2, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Here are the exposure-exposure correlations learned by RSparsePro:

```{r exp_exp_cor_rsparsepro, echo=FALSE}
data_rsparsepro2 <- melt(R_rsparsepro[1:33, 1:33])
ggplot(data_rsparsepro2, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```


And here are the exposure-exposure correlations with weights = betas:

```{r exp_exp_cor_nolearnwts, echo=FALSE}
data_nolearnwts2 <- melt(R_nolearnwts[1:33, 1:33])
ggplot(data_nolearnwts2, aes(x = Var2, y = Var1, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(x = "", y = "", fill = "Value") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```



# Results with lipoprotein subfractions

In this analysis, I analyzed the causal effects of 94 lipoprotein subfraction
("met-c" on IEU GWAS Browser) traits on the same four outcome traits.
Here we are moving away from complex trait exposures with (sometimes) known
relationships to the outcome, and into a setting with many very highly correlated
molecular exposures.

The processing pipeline is similar to the previous one, and the highly correlated
traits are mostly pretty obvious (e.g. LDL subfractions are highly correlated
with each other, and not with HDL subfractions), so I just list the traits and
then give the results below.


### Trait names

```{r metc-exposures}
# read in data and select methods to display
dat_metc = fread('data/summary_of_results_metc_noannih_clumped.tsv', sep='\t', header=T)
dat_metc = dat_metc %>% select(Exposure, Outcome, famr_none, famr_gfa, ivw, robust, ivw.univar)
# sort by average PIP across all methods
dat_metc$Average = rowSums(dat_metc %>% select(famr_none, famr_gfa, ivw, robust, ivw.univar)) / (ncol(dat_metc) - 2)
dat_metc = dat_metc[order(dat_metc$Average, decreasing=TRUE), ]
dat_metc$Average = NULL 
sort(unique(dat_metc$Exposure))
```

### Results

One important note: because the GWAS loci for different lipoprotein subfractions
overlap highly, there are only 51 total loci compared to 94 exposures. 
Thus, for standard MVMR methods, SNPs cannot be selected by the "SNP with max Z
for each locus" method, as this would result in fewer instruments than exposures.
Thus, for these methods, we have to use the LD clumping approach to select 
instruments, which, as shown above, can lead to some false positives.
We see similar results below.

Also, FAMR with factors regressed out doesn't yield any credible sets containing
any exposures. So we show the FAMR results with GFA factors not regressed out.

The following are the main takeaways so far:

* Multivariable MR methods rarely identify any individual exposure-outcome pair
as significant, which is to be expected given the very high correlations between
the exposures.

* However, FAMR_GFA does identify a credible set containing many LDL-related
exposures for CAD, which is a positive sign. This set contains: Apolipoprotein B,
Cholesterol esters in large VLDL, Total lipids in large LDL, 
Concentration of large LDL particles, Total cholesterol in LDL,
Total cholesterol in medium LDL, Cholesterol esters in medium LDL,
Total lipids in medium LDL, Concentration of medium LDL particles,
Phospholipids in medium LDL, Total cholesterol in small LDL,
and Total cholesterol in small VLDL.

* FAMR does not identify credible sets containing exposures for any other outcome trait.

* The exposures with the highest PIPs for MVMR-IVW and MVMR-Robust are mostly 
unknown (effect of IDL and LDL subfractions on T2D) or unlikely to be true
(effect of IDL and LDL subfractions on Asthma).

* Meanwhile, univariate IVW finds many individual significant exposure-outcome pairs,
mostly LDL-related and VLDL-related exposures on CAD and Stroke, as well as ApoB,
which makes sense. However, it also identifies some HDL subfractions as having
an effect on CAD and/or Stroke, which are probably false positives.

Overall the FAMR results are a bit conservative but it avoids likely false 
positives found by other methods and provides some useful qualitative information via
the credible sets.

Notably, there is 
[another study](https://www.ahajournals.org/doi/10.1161/CIRCULATIONAHA.121.053797)
on the effects of lipoprotein subfractions on CAD risk.
That study found that large low-density lipoprotein particle concentration was 
the most likely subfraction to affect CAD, while ApoB was the most likely causal
lipoprotein-related risk factor.
In agreement with this, FAMR_GFA's credible set contains both of these traits,
although their PIPs are not among the highest few in the set.
The highest FAMR PIPs in the set are for "Total cholesterol in LDL" followed by
"Total cholesterol in medium LDL".
Their study approached instrument selection differently and did not use the
same set of exposures, so the results are not directly comparable, but this is 
an encouraging sign.


<details><summary>Click to expand</summary>

```{r print_results_metc}
knitr::kable(dat_metc)
```

</details>

